// @fx-lib/components/esm2015/lib/fx-security/guards/auth-guard.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { AuthenticationService } from '../services/authentication.service';
import { AuthorizationService } from '../services/authorization.service';
/** @type {?} */
const NO_PERMISSION_ROUTE = 'no-permission';
export class AuthGuard {
    /**
     * @param {?} authenticationService
     * @param {?} authorizationService
     * @param {?} router
     */
    constructor(authenticationService, authorizationService, router) {
        this.authenticationService = authenticationService;
        this.authorizationService = authorizationService;
        this.router = router;
    }
    /**
     * @param {?} route
     * @param {?} state
     * @return {?}
     */
    canActivate(route, state) {
        return this.checkLogin(state.url) && this.checkRoles(route.data) && this.checkBoundaryValue(route.data, route.paramMap);
    }
    /**
     * @private
     * @param {?} url
     * @return {?}
     */
    checkLogin(url) {
        if (this.authenticationService.isLoggedIn) {
            return true;
        }
        this.authenticationService.redirectToIdpLogin(url);
        return false;
    }
    /**
     * @private
     * @param {?} data
     * @param {?} paramMap
     * @return {?}
     */
    checkBoundaryValue(data, paramMap) {
        /** @type {?} */
        let hasPermission = false;
        if (!data || !data.boundaryParamName) {
            return true;
        }
        if (this.authorizationService.hasDevopsRole()) {
            return true;
        }
        this.authenticationService.getPayload().roles.forEach((/**
         * @param {?} role
         * @return {?}
         */
        role => {
            if (this.authorizationService.hasRoleForBoundary(role.role, paramMap.get(data.boundaryParamName))) {
                hasPermission = true;
                return;
            }
        }));
        if (!hasPermission) {
            this.router.navigateByUrl(NO_PERMISSION_ROUTE);
        }
        return hasPermission;
    }
    /**
     * @private
     * @param {?} data
     * @return {?}
     */
    checkRoles(data) {
        /** @type {?} */
        let hasPermission = false;
        if (!data || !data.roles) {
            return true;
        }
        data.roles.forEach((/**
         * @param {?} role
         * @return {?}
         */
        role => {
            /** @type {?} */
            const requiredRoleName = role;
            /** @type {?} */
            const hasRole = this.authorizationService.hasRole(requiredRoleName);
            if (hasRole) {
                hasPermission = true;
                return;
            }
        }));
        if (!hasPermission) {
            this.router.navigateByUrl(NO_PERMISSION_ROUTE);
        }
        return hasPermission;
    }
}
AuthGuard.decorators = [
    { type: Injectable }
];
/** @nocollapse */
AuthGuard.ctorParameters = () => [
    { type: AuthenticationService },
    { type: AuthorizationService },
    { type: Router }
];
if (false) {
    /**
     * @type {?}
     * @private
     */
    AuthGuard.prototype.authenticationService;
    /**
     * @type {?}
     * @private
     */
    AuthGuard.prototype.authorizationService;
    /**
     * @type {?}
     * @private
     */
    AuthGuard.prototype.router;
}
