// @fx-lib/components/esm5/lib/shared/helper/local-storage.helper.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { DEBUG, FACILITY } from '../constants/local-storage.constants';
/**
 * @param {?} key
 * @param {?} value
 * @return {?}
 */
export function setItem(key, value) {
    if (!key) {
        return;
    }
    localStorage.setItem(key, value);
}
/**
 * @param {?} key
 * @return {?}
 */
export function deleteItem(key) {
    if (!key) {
        return;
    }
    localStorage.removeItem(key);
}
/**
 * @param {?} key
 * @return {?}
 */
export function getItem(key) {
    if (!key) {
        return null;
    }
    return localStorage.getItem(key);
}
/**
 * @param {?} facility
 * @return {?}
 */
export function setFacility(facility) {
    setItem(FACILITY, facility);
}
/**
 * @return {?}
 */
export function getFacility() {
    return getItem(FACILITY);
}
/**
 * @return {?}
 */
export function deleteFacility() {
    deleteItem(FACILITY);
}
/**
 * @param {?} debug
 * @return {?}
 */
export function setDebugMode(debug) {
    setItem(DEBUG, debug ? '1' : '0');
}
/**
 * @return {?}
 */
export function isDebugMode() {
    return getItem(DEBUG) === '1';
}
/**
 * @return {?}
 */
export function deleteDebugMode() {
    deleteItem(DEBUG);
}
