// @fx-lib/components/esm5/lib/fx-security/components/fx-login/fx-login.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { ChangeDetectorRef, Component, Input } from '@angular/core';
import { Subscription } from 'rxjs';
import { AuthenticationService } from '../../services/authentication.service';
import { UserService } from '../../services/user.service';
import { faUser } from '@fortawesome/free-solid-svg-icons/faUser';
var FxLoginComponent = /** @class */ (function () {
    function FxLoginComponent(authenticationService, userService, changeDetector) {
        this.authenticationService = authenticationService;
        this.userService = userService;
        this.changeDetector = changeDetector;
        this.btnStyle = '';
        this.dropDownStyling = '';
        this.languageSelectorRadioGroupStyling = 'fx-styling';
        this.icon = 'fa-user';
        this.fasUser = faUser;
        this.subscriptions = new Subscription();
    }
    /**
     * @return {?}
     */
    FxLoginComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.subscriptions.add(this.authenticationService.loginEvents.subscribe((/**
         * @return {?}
         */
        function () {
            _this.username = _this.userService.getUsername();
            _this.changeDetector.markForCheck();
        })));
    };
    /**
     * @return {?}
     */
    FxLoginComponent.prototype.isNotLoggedIn = /**
     * @return {?}
     */
    function () {
        return !this.username || this.username.trim().length === 0;
    };
    /**
     * @return {?}
     */
    FxLoginComponent.prototype.login = /**
     * @return {?}
     */
    function () {
        this.authenticationService.loginWithRedirectUrl();
    };
    /**
     * @return {?}
     */
    FxLoginComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.subscriptions.unsubscribe();
    };
    /**
     * @param {?=} selectedIcon
     * @return {?}
     */
    FxLoginComponent.prototype.setIcon = /**
     * @param {?=} selectedIcon
     * @return {?}
     */
    function (selectedIcon) {
        switch (selectedIcon) {
            case 'fasUser': {
                return this.fasUser;
            }
            default: {
                return this.fasUser;
            }
        }
    };
    FxLoginComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-fx-login',
                    template: "<div [ngClass]=\"dropDownStyling\">\r\n  <div ngbDropdown class=\"dropdown\">\r\n    <button [ngClass]=\"['btn', btnStyle]\" type=\"button\" id=\"dropdownLoginButton\" ngbDropdownToggle>\r\n      <fa-icon *ngIf=\"icon\" [icon]=\"setIcon(icon)\"></fa-icon>\r\n      <a *ngIf=\"icon\">&nbsp;</a>\r\n      {{ !isNotLoggedIn() ? username : 'i18n.fx-components.security.login' | translate}}\r\n    </button>\r\n\r\n    <div ngbDropdownMenu class=\"dropdown-menu\" aria-labelledby=\"dropdownLoginButton\">\r\n      <div>\r\n        <app-fx-language-selector [radioGroupStyling]=\"languageSelectorRadioGroupStyling\"></app-fx-language-selector>\r\n      </div>\r\n      <div class=\"dropdown-divider\"></div>\r\n      <button ngbDropdownItem class=\"dropdown-item\" appLogout=\"\" *ngIf=\"!isNotLoggedIn()\">\r\n        {{'i18n.fx-components.security.logout' | translate}}\r\n      </button>\r\n      <button ngbDropdownItem class=\"dropdown-item\" (click)=\"login()\" *ngIf=\"isNotLoggedIn()\">\r\n        {{'i18n.fx-components.security.login' | translate}}\r\n      </button>\r\n    </div>\r\n  </div>\r\n</div>",
                    styles: [".login-icon{padding-right:4px}.fx-styling .no-permission-text{color:#fff}.fx-styling .btn,.fx-styling .btn:hover{color:#fff;background-color:transparent}.fx-styling .btn:focus{box-shadow:none}.fx-styling .dropdown-item:active{background-color:#004666}"]
                }] }
    ];
    /** @nocollapse */
    FxLoginComponent.ctorParameters = function () { return [
        { type: AuthenticationService },
        { type: UserService },
        { type: ChangeDetectorRef }
    ]; };
    FxLoginComponent.propDecorators = {
        btnStyle: [{ type: Input }],
        dropDownStyling: [{ type: Input }],
        languageSelectorRadioGroupStyling: [{ type: Input }],
        icon: [{ type: Input }]
    };
    return FxLoginComponent;
}());
export { FxLoginComponent };
if (false) {
    /** @type {?} */
    FxLoginComponent.prototype.btnStyle;
    /** @type {?} */
    FxLoginComponent.prototype.dropDownStyling;
    /** @type {?} */
    FxLoginComponent.prototype.languageSelectorRadioGroupStyling;
    /** @type {?} */
    FxLoginComponent.prototype.icon;
    /** @type {?} */
    FxLoginComponent.prototype.fasUser;
    /** @type {?} */
    FxLoginComponent.prototype.username;
    /**
     * @type {?}
     * @private
     */
    FxLoginComponent.prototype.subscriptions;
    /**
     * @type {?}
     * @private
     */
    FxLoginComponent.prototype.authenticationService;
    /**
     * @type {?}
     * @private
     */
    FxLoginComponent.prototype.userService;
    /**
     * @type {?}
     * @private
     */
    FxLoginComponent.prototype.changeDetector;
}
