// @fx-lib/components/esm2015/lib/shared/api-client/interceptors/debug-data-interceptor.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { AuthenticationService } from '../../../fx-security/services/authentication.service';
import { isDebugMode } from '../../helper/local-storage.helper';
export class DebugDataInterceptor {
    /**
     * @param {?} authService
     * @param {?} route
     */
    constructor(authService, route) {
        this.authService = authService;
        this.route = route;
    }
    /**
     * interceptor for debug data instead of real data
     * [example:
     *  api call https://fx-part-demand-gateway-service-integration/ccs/api/v1/11/vehicle/
     *  redirects to ./assets/mock-data/fx-part-demand-gateway-service-integration/ccs/api/v1/11/vehicle.json
     * ]
     * @param {?} request
     * @param {?} next
     * @return {?}
     */
    intercept(request, next) {
        /** @type {?} */
        const queryParams = this.route.snapshot.queryParams;
        /** @type {?} */
        const debug = (queryParams['debug'] === 'true' || isDebugMode()) && queryParams['debug'] !== 'false';
        /** @type {?} */
        const isLoggedIn = this.authService.isLoggedIn;
        /** @type {?} */
        let newRequest = request;
        /** @type {?} */
        const searchString = '.apps.emea.vwapps.io';
        /** @type {?} */
        const index = request.url.indexOf(searchString);
        if (isLoggedIn && debug && index > -1) {
            /** @type {?} */
            const apiName = request.url.substring(request.url.indexOf('//') + 2, index);
            /** @type {?} */
            let newUrl = request.url.substring(index + searchString.length).replace(/\/\s*$/, '');
            newUrl = `./assets/mock-data/${apiName}${newUrl}.json`;
            newRequest = request.clone({
                url: newUrl,
                method: 'Get',
            });
        }
        return next.handle(newRequest);
    }
}
DebugDataInterceptor.decorators = [
    { type: Injectable }
];
/** @nocollapse */
DebugDataInterceptor.ctorParameters = () => [
    { type: AuthenticationService },
    { type: ActivatedRoute }
];
if (false) {
    /**
     * @type {?}
     * @private
     */
    DebugDataInterceptor.prototype.authService;
    /**
     * @type {?}
     * @private
     */
    DebugDataInterceptor.prototype.route;
}
