// @fx-lib/components/esm2015/lib/shared/api-client/interceptors/http-token-interceptor.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { AuthenticationService } from '../../../fx-security/services/authentication.service';
export class HttpTokenInterceptor {
    /**
     * @param {?} authService
     */
    constructor(authService) {
        this.authService = authService;
    }
    /**
     * @param {?} request
     * @param {?} next
     * @return {?}
     */
    intercept(request, next) {
        /** @type {?} */
        const bearerToken = this.authService.getAuthTokenHeaderString();
        /** @type {?} */
        const isLoggedIn = this.authService.isLoggedIn;
        /** @type {?} */
        let newRequest = request;
        if (isLoggedIn && bearerToken && request.url.indexOf('.apps.emea.vwapps.io') > -1) {
            /** @type {?} */
            let url = newRequest.url;
            url = this.replaceHttpOrPrefixUrlWithHttps(url);
            newRequest = request.clone({
                setHeaders: {
                    Authorization: bearerToken,
                },
                url: url,
            });
        }
        return next.handle(newRequest);
    }
    /**
     * @private
     * @param {?} originalUrl
     * @return {?}
     */
    replaceHttpOrPrefixUrlWithHttps(originalUrl) {
        /** @type {?} */
        let url = originalUrl;
        if (!url.startsWith('https')) {
            if (url.startsWith('http:')) {
                url = url.replace('http:', 'https:');
            }
            else if (url.startsWith('//')) {
                url = 'https:' + url;
            }
            else {
                url = 'https://' + url;
            }
        }
        return url;
    }
}
HttpTokenInterceptor.decorators = [
    { type: Injectable }
];
/** @nocollapse */
HttpTokenInterceptor.ctorParameters = () => [
    { type: AuthenticationService }
];
if (false) {
    /**
     * @type {?}
     * @private
     */
    HttpTokenInterceptor.prototype.authService;
}
