// @fx-lib/components/esm5/lib/fx-security/directives/is-logged-in.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { ChangeDetectorRef, Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
import { AuthenticationService } from '../services/authentication.service';
var IsLoggedInDirective = /** @class */ (function () {
    function IsLoggedInDirective(authService, _viewContainer, templateRef, changeDetector) {
        var _this = this;
        this.authService = authService;
        this._viewContainer = _viewContainer;
        this.changeDetector = changeDetector;
        this._context = new IsLoggedInContext();
        this._thenTemplateRef = null;
        this._elseTemplateRef = null;
        this._thenViewRef = null;
        this._elseViewRef = null;
        this._thenTemplateRef = templateRef;
        this.loginEventsSubscription = authService.loginEvents.subscribe((/**
         * @return {?}
         */
        function () {
            _this._updateView();
            _this.changeDetector.markForCheck();
        }));
    }
    /**
     * @return {?}
     */
    IsLoggedInDirective.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        if (this.loginEventsSubscription) {
            this.loginEventsSubscription.unsubscribe();
        }
    };
    Object.defineProperty(IsLoggedInDirective.prototype, "hasRole", {
        set: /**
         * @param {?} condition
         * @return {?}
         */
        function (condition) {
            this._context.$implicit = condition;
            this._updateView();
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IsLoggedInDirective.prototype, "hasRoleThen", {
        set: /**
         * @param {?} templateRef
         * @return {?}
         */
        function (templateRef) {
            this._thenTemplateRef = templateRef;
            this._thenViewRef = null; // clear previous view if any.
            this._updateView();
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IsLoggedInDirective.prototype, "hasRoleElse", {
        set: /**
         * @param {?} templateRef
         * @return {?}
         */
        function (templateRef) {
            this._elseTemplateRef = templateRef;
            this._elseViewRef = null; // clear previous view if any.
            this._updateView();
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @private
     * @return {?}
     */
    IsLoggedInDirective.prototype._updateView = /**
     * @private
     * @return {?}
     */
    function () {
        if (this.authService.isLoggedIn) {
            if (!this._thenViewRef) {
                this._viewContainer.clear();
                this._elseViewRef = null;
                if (this._thenTemplateRef) {
                    this._thenViewRef =
                        this._viewContainer.createEmbeddedView(this._thenTemplateRef, this._context);
                }
            }
        }
        else {
            if (!this._elseViewRef) {
                this._viewContainer.clear();
                this._thenViewRef = null;
                if (this._elseTemplateRef) {
                    this._elseViewRef =
                        this._viewContainer.createEmbeddedView(this._elseTemplateRef, this._context);
                }
            }
        }
    };
    IsLoggedInDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[appIsLoggedIn]'
                },] }
    ];
    /** @nocollapse */
    IsLoggedInDirective.ctorParameters = function () { return [
        { type: AuthenticationService },
        { type: ViewContainerRef },
        { type: TemplateRef },
        { type: ChangeDetectorRef }
    ]; };
    IsLoggedInDirective.propDecorators = {
        hasRole: [{ type: Input }],
        hasRoleThen: [{ type: Input }],
        hasRoleElse: [{ type: Input }]
    };
    return IsLoggedInDirective;
}());
export { IsLoggedInDirective };
if (false) {
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype._context;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype._thenTemplateRef;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype._elseTemplateRef;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype._thenViewRef;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype._elseViewRef;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype.loginEventsSubscription;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype.authService;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype._viewContainer;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype.changeDetector;
}
var IsLoggedInContext = /** @class */ (function () {
    function IsLoggedInContext() {
        this.$implicit = null;
    }
    return IsLoggedInContext;
}());
export { IsLoggedInContext };
if (false) {
    /** @type {?} */
    IsLoggedInContext.prototype.$implicit;
}
