// @fx-lib/components/esm5/lib/shared/interfaces/datatable-column.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @record
 */
export function DatatableColumnTooltip() { }
if (false) {
    /** @type {?|undefined} */
    DatatableColumnTooltip.prototype.column;
    /** @type {?|undefined} */
    DatatableColumnTooltip.prototype.container;
    /** @type {?|undefined} */
    DatatableColumnTooltip.prototype.placement;
    /** @type {?|undefined} */
    DatatableColumnTooltip.prototype.showCondition;
    /** @type {?|undefined} */
    DatatableColumnTooltip.prototype.text;
}
/**
 * @record
 */
export function DatatableColumnIcon() { }
if (false) {
    /** @type {?} */
    DatatableColumnIcon.prototype.class;
    /** @type {?|undefined} */
    DatatableColumnIcon.prototype.placement;
}
/**
 * @record
 */
export function DatatableColumnSubIcon() { }
if (false) {
    /** @type {?} */
    DatatableColumnSubIcon.prototype.class;
    /** @type {?|undefined} */
    DatatableColumnSubIcon.prototype.placement;
}
/**
 * @record
 */
export function DatatableColumnBorder() { }
if (false) {
    /** @type {?|undefined} */
    DatatableColumnBorder.prototype.bottom;
    /** @type {?|undefined} */
    DatatableColumnBorder.prototype.left;
    /** @type {?|undefined} */
    DatatableColumnBorder.prototype.right;
    /** @type {?|undefined} */
    DatatableColumnBorder.prototype.top;
}
/**
 * @record
 */
export function DatatableColumnNavigationRoute() { }
if (false) {
    /** @type {?} */
    DatatableColumnNavigationRoute.prototype.routerLink;
    /** @type {?|undefined} */
    DatatableColumnNavigationRoute.prototype.queryParams;
}
/**
 * @record
 */
export function DatatableColumnFoldColumns() { }
if (false) {
    /** @type {?} */
    DatatableColumnFoldColumns.prototype.isFolded;
    /** @type {?} */
    DatatableColumnFoldColumns.prototype.columnNames;
}
/**
 * @record
 */
export function DatatableColumn() { }
if (false) {
    /** @type {?|undefined} */
    DatatableColumn.prototype.cellAlign;
    /** @type {?|undefined} */
    DatatableColumn.prototype.cellBorder;
    /** @type {?|undefined} */
    DatatableColumn.prototype.colorRanges;
    /** @type {?|undefined} */
    DatatableColumn.prototype.format;
    /** @type {?|undefined} */
    DatatableColumn.prototype.isFolded;
    /** @type {?|undefined} */
    DatatableColumn.prototype.foldColumns;
    /** @type {?|undefined} */
    DatatableColumn.prototype.headerAlign;
    /** @type {?|undefined} */
    DatatableColumn.prototype.headerBorder;
    /** @type {?|undefined} */
    DatatableColumn.prototype.headerIcon;
    /** @type {?|undefined} */
    DatatableColumn.prototype.headerSubIcon;
    /** @type {?|undefined} */
    DatatableColumn.prototype.noTextOverflow;
    /** @type {?|undefined} */
    DatatableColumn.prototype.navigationRoute;
    /** @type {?|undefined} */
    DatatableColumn.prototype.selectable;
    /** @type {?|undefined} */
    DatatableColumn.prototype.staticCellTooltip;
    /** @type {?|undefined} */
    DatatableColumn.prototype.staticHeaderTooltip;
    /** @type {?|undefined} */
    DatatableColumn.prototype.translationKey;
    /** @type {?|undefined} */
    DatatableColumn.prototype.translationValues;
    /** @type {?|undefined} */
    DatatableColumn.prototype.uniqueCellClass;
}
