// @fx-lib/components/esm2015/lib/fx-header/services/fx-header-config.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import { FxHeaderConfigValidatorService } from './fx-header-config-validator.service';
export class FxHeaderConfigService {
    /**
     * @param {?} fxheaderConfigValidatorService
     */
    constructor(fxheaderConfigValidatorService) {
        this.fxheaderConfigValidatorService = fxheaderConfigValidatorService;
        this.config = new BehaviorSubject({
            appTitleShort: '',
            appTitle: '',
            routes: [],
            staticComponents: [],
            components: []
        });
    }
    /**
     * @param {?} config
     * @return {?}
     */
    setConfig(config) {
        if (!this.fxheaderConfigValidatorService.validate(config)) {
            return;
        }
        this.config.next(config);
    }
}
FxHeaderConfigService.decorators = [
    { type: Injectable }
];
/** @nocollapse */
FxHeaderConfigService.ctorParameters = () => [
    { type: FxHeaderConfigValidatorService }
];
if (false) {
    /** @type {?} */
    FxHeaderConfigService.prototype.config;
    /**
     * @type {?}
     * @private
     */
    FxHeaderConfigService.prototype.fxheaderConfigValidatorService;
}
