// @fx-lib/components/esm5/lib/shared/api-client/interceptors/debug-data-interceptor.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { AuthenticationService } from '../../../fx-security/services/authentication.service';
import { isDebugMode } from '../../helper/local-storage.helper';
var DebugDataInterceptor = /** @class */ (function () {
    function DebugDataInterceptor(authService, route) {
        this.authService = authService;
        this.route = route;
    }
    /**
     * interceptor for debug data instead of real data
     * [example:
     *  api call https://fx-part-demand-gateway-service-integration/ccs/api/v1/11/vehicle/
     *  redirects to ./assets/mock-data/fx-part-demand-gateway-service-integration/ccs/api/v1/11/vehicle.json
     * ]
     */
    /**
     * interceptor for debug data instead of real data
     * [example:
     *  api call https://fx-part-demand-gateway-service-integration/ccs/api/v1/11/vehicle/
     *  redirects to ./assets/mock-data/fx-part-demand-gateway-service-integration/ccs/api/v1/11/vehicle.json
     * ]
     * @param {?} request
     * @param {?} next
     * @return {?}
     */
    DebugDataInterceptor.prototype.intercept = /**
     * interceptor for debug data instead of real data
     * [example:
     *  api call https://fx-part-demand-gateway-service-integration/ccs/api/v1/11/vehicle/
     *  redirects to ./assets/mock-data/fx-part-demand-gateway-service-integration/ccs/api/v1/11/vehicle.json
     * ]
     * @param {?} request
     * @param {?} next
     * @return {?}
     */
    function (request, next) {
        /** @type {?} */
        var queryParams = this.route.snapshot.queryParams;
        /** @type {?} */
        var debug = (queryParams['debug'] === 'true' || isDebugMode()) && queryParams['debug'] !== 'false';
        /** @type {?} */
        var isLoggedIn = this.authService.isLoggedIn;
        /** @type {?} */
        var newRequest = request;
        /** @type {?} */
        var searchString = '.apps.emea.vwapps.io';
        /** @type {?} */
        var index = request.url.indexOf(searchString);
        if (isLoggedIn && debug && index > -1) {
            /** @type {?} */
            var apiName = request.url.substring(request.url.indexOf('//') + 2, index);
            /** @type {?} */
            var newUrl = request.url.substring(index + searchString.length).replace(/\/\s*$/, '');
            newUrl = "./assets/mock-data/" + apiName + newUrl + ".json";
            newRequest = request.clone({
                url: newUrl,
                method: 'Get',
            });
        }
        return next.handle(newRequest);
    };
    DebugDataInterceptor.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    DebugDataInterceptor.ctorParameters = function () { return [
        { type: AuthenticationService },
        { type: ActivatedRoute }
    ]; };
    return DebugDataInterceptor;
}());
export { DebugDataInterceptor };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DebugDataInterceptor.prototype.authService;
    /**
     * @type {?}
     * @private
     */
    DebugDataInterceptor.prototype.route;
}
