// @fx-lib/components/esm5/lib/shared/api-client/interceptors/http-token-interceptor.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { AuthenticationService } from '../../../fx-security/services/authentication.service';
var HttpTokenInterceptor = /** @class */ (function () {
    function HttpTokenInterceptor(authService) {
        this.authService = authService;
    }
    /**
     * @param {?} request
     * @param {?} next
     * @return {?}
     */
    HttpTokenInterceptor.prototype.intercept = /**
     * @param {?} request
     * @param {?} next
     * @return {?}
     */
    function (request, next) {
        /** @type {?} */
        var bearerToken = this.authService.getAuthTokenHeaderString();
        /** @type {?} */
        var isLoggedIn = this.authService.isLoggedIn;
        /** @type {?} */
        var newRequest = request;
        if (isLoggedIn && bearerToken && request.url.indexOf('.apps.emea.vwapps.io') > -1) {
            /** @type {?} */
            var url = newRequest.url;
            url = this.replaceHttpOrPrefixUrlWithHttps(url);
            newRequest = request.clone({
                setHeaders: {
                    Authorization: bearerToken,
                },
                url: url,
            });
        }
        return next.handle(newRequest);
    };
    /**
     * @private
     * @param {?} originalUrl
     * @return {?}
     */
    HttpTokenInterceptor.prototype.replaceHttpOrPrefixUrlWithHttps = /**
     * @private
     * @param {?} originalUrl
     * @return {?}
     */
    function (originalUrl) {
        /** @type {?} */
        var url = originalUrl;
        if (!url.startsWith('https')) {
            if (url.startsWith('http:')) {
                url = url.replace('http:', 'https:');
            }
            else if (url.startsWith('//')) {
                url = 'https:' + url;
            }
            else {
                url = 'https://' + url;
            }
        }
        return url;
    };
    HttpTokenInterceptor.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    HttpTokenInterceptor.ctorParameters = function () { return [
        { type: AuthenticationService }
    ]; };
    return HttpTokenInterceptor;
}());
export { HttpTokenInterceptor };
if (false) {
    /**
     * @type {?}
     * @private
     */
    HttpTokenInterceptor.prototype.authService;
}
