// @fx-lib/components/esm5/lib/fx-security/services/authorization.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable, Inject } from '@angular/core';
import { AuthenticationService } from './authentication.service';
import { SECURITY_CONFIG } from '../domain/security-config';
var AuthorizationService = /** @class */ (function () {
    function AuthorizationService(authenticationService, securityConfig) {
        this.authenticationService = authenticationService;
        this.securityConfig = securityConfig;
    }
    /**
     * @return {?}
     */
    AuthorizationService.prototype.hasViewRole = /**
     * @return {?}
     */
    function () {
        return this.checkMultipleRoles(this.securityConfig.roles.view);
    };
    /**
     * @param {?} boundaryValue
     * @return {?}
     */
    AuthorizationService.prototype.hasViewRoleForBoundary = /**
     * @param {?} boundaryValue
     * @return {?}
     */
    function (boundaryValue) {
        return this.checkMultipleRolesForBoundary(this.securityConfig.roles.view, boundaryValue);
    };
    /**
     * @return {?}
     */
    AuthorizationService.prototype.hasEditRole = /**
     * @return {?}
     */
    function () {
        return this.checkMultipleRoles(this.securityConfig.roles.edit);
    };
    /**
     * @param {?} boundaryValue
     * @return {?}
     */
    AuthorizationService.prototype.hasEditRoleForBoundary = /**
     * @param {?} boundaryValue
     * @return {?}
     */
    function (boundaryValue) {
        return this.checkMultipleRolesForBoundary(this.securityConfig.roles.edit, boundaryValue);
    };
    /**
     * @return {?}
     */
    AuthorizationService.prototype.hasDevopsRole = /**
     * @return {?}
     */
    function () {
        return this.checkMultipleRoles(this.securityConfig.roles.devops);
    };
    /**
     * @param {?} roleToCheck
     * @return {?}
     */
    AuthorizationService.prototype.hasRole = /**
     * @param {?} roleToCheck
     * @return {?}
     */
    function (roleToCheck) {
        /** @type {?} */
        var payload = this.authenticationService.getPayload();
        return payload.roles.some((/**
         * @param {?} role
         * @return {?}
         */
        function (role) {
            return role.role === roleToCheck;
        }));
    };
    /**
     * @param {?} roleToCheck
     * @param {?} boundaryValue
     * @return {?}
     */
    AuthorizationService.prototype.hasRoleForBoundary = /**
     * @param {?} roleToCheck
     * @param {?} boundaryValue
     * @return {?}
     */
    function (roleToCheck, boundaryValue) {
        /** @type {?} */
        var payload = this.authenticationService.getPayload();
        /** @type {?} */
        var foundRole = payload.roles.find((/**
         * @param {?} r
         * @return {?}
         */
        function (r) { return r.role === roleToCheck; }));
        if (!foundRole) {
            return false;
        }
        /** @type {?} */
        var devopsRole = this.securityConfig.roles.devops.find((/**
         * @param {?} r
         * @return {?}
         */
        function (r) { return r === foundRole.role; }));
        if (devopsRole) {
            return true;
        }
        /** @type {?} */
        var boundaries = foundRole.boundaries[this.securityConfig.boundary_name];
        return boundaries.some((/**
         * @param {?} boundary
         * @return {?}
         */
        function (boundary) { return boundary === boundaryValue; }));
    };
    /**
     * @private
     * @param {?} roles
     * @return {?}
     */
    AuthorizationService.prototype.checkMultipleRoles = /**
     * @private
     * @param {?} roles
     * @return {?}
     */
    function (roles) {
        var _this = this;
        return roles.some((/**
         * @param {?} role
         * @return {?}
         */
        function (role) { return _this.hasRole(role); }));
    };
    /**
     * @private
     * @param {?} roles
     * @param {?} boundaryValue
     * @return {?}
     */
    AuthorizationService.prototype.checkMultipleRolesForBoundary = /**
     * @private
     * @param {?} roles
     * @param {?} boundaryValue
     * @return {?}
     */
    function (roles, boundaryValue) {
        var _this = this;
        return roles.some((/**
         * @param {?} role
         * @return {?}
         */
        function (role) { return _this.hasRoleForBoundary(role, boundaryValue); }));
    };
    AuthorizationService.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    AuthorizationService.ctorParameters = function () { return [
        { type: AuthenticationService },
        { type: undefined, decorators: [{ type: Inject, args: [SECURITY_CONFIG,] }] }
    ]; };
    return AuthorizationService;
}());
export { AuthorizationService };
if (false) {
    /** @type {?} */
    AuthorizationService.prototype.securityConfig;
    /**
     * @type {?}
     * @private
     */
    AuthorizationService.prototype.authenticationService;
}
