// @fx-lib/components/esm5/lib/fx-security/services/login.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Inject, Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { throwError } from 'rxjs';
import { catchError, map } from 'rxjs/operators';
import { SECURITY_CONFIG } from '../domain/security-config';
var LoginService = /** @class */ (function () {
    function LoginService(http, securityConfig) {
        this.http = http;
        this.ursUrl = 'api/v1/auth';
        this.securityConfig = securityConfig;
    }
    /**
     * @param {?} idToken
     * @return {?}
     */
    LoginService.prototype.login = /**
     * @param {?} idToken
     * @return {?}
     */
    function (idToken) {
        var _this = this;
        /** @type {?} */
        var url = this.securityConfig.authorization_service_url + this.ursUrl;
        return this.http
            .post(url, idToken, { observe: 'response', responseType: 'text' })
            .pipe(map((/**
         * @param {?} response
         * @return {?}
         */
        function (response) {
            return response.body;
        })), catchError((/**
         * @param {?} error
         * @return {?}
         */
        function (error) {
            return _this.handleError(error);
        })));
    };
    /**
     * @private
     * @param {?} error
     * @return {?}
     */
    LoginService.prototype.handleError = /**
     * @private
     * @param {?} error
     * @return {?}
     */
    function (error) {
        /** @type {?} */
        var errorMessage = error.message ? error.message : error.toString();
        console.error(errorMessage);
        return throwError(errorMessage);
    };
    LoginService.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    LoginService.ctorParameters = function () { return [
        { type: HttpClient },
        { type: undefined, decorators: [{ type: Inject, args: [SECURITY_CONFIG,] }] }
    ]; };
    return LoginService;
}());
export { LoginService };
if (false) {
    /** @type {?} */
    LoginService.prototype.securityConfig;
    /**
     * @type {?}
     * @private
     */
    LoginService.prototype.ursUrl;
    /**
     * @type {?}
     * @private
     */
    LoginService.prototype.http;
}
