// @fx-lib/components/esm2015/lib/fx-confidentiality-logo/fx-confidentiality-logo.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
/**
 * @record
 */
export function ConfidentialityLogo() { }
if (false) {
    /** @type {?} */
    ConfidentialityLogo.prototype.type;
    /** @type {?} */
    ConfidentialityLogo.prototype.path;
}
export class FxConfidentialityLogoComponent {
    constructor() {
        this.confidentialityLogo = [
            { type: 'public', path: '/assets/images/confidentiality_logos/VWAG_Public.svg' },
            { type: 'public_scaled', path: '/assets/images/confidentiality_logos/VWAG_Public_scaled.svg' },
            { type: 'internal', path: '/assets/images/confidentiality_logos/VWAG_Internal.svg' },
            { type: 'internal_scaled', path: '/assets/images/confidentiality_logos/VWAG_Internal_scaled.svg' },
            { type: 'confidential', path: '/assets/images/confidentiality_logos/VWAG_Confidential.svg' },
            { type: 'confidential_scaled', path: '/assets/images/confidentiality_logos/VWAG_Confidential_scaled.svg' },
            { type: 'secret', path: '/assets/images/confidentiality_logos/VWAG_Secret.svg' },
            { type: 'secret_scaled', path: '/assets/images/confidentiality_logos/VWAG_Secret_scaled.svg' },
        ];
        this.logoType = 'internal';
        this.logoHeight = 60;
    }
    /**
     * @return {?}
     */
    getPath() {
        return this.confidentialityLogo.find((/**
         * @param {?} logo
         * @return {?}
         */
        logo => logo.type === this.logoType)).path;
    }
}
FxConfidentialityLogoComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-fx-confidentiality-logo',
                template: "<img [ngClass]=\"logoColor\" [src]=\"getPath()\" alt=\"confidentiality logo\" [height]=\"logoHeight\">\r\n",
                styles: [".white{-webkit-filter:invert(1);filter:invert(1)}.grey{-webkit-filter:invert(.3);filter:invert(.3)}"]
            }] }
];
/** @nocollapse */
FxConfidentialityLogoComponent.ctorParameters = () => [];
FxConfidentialityLogoComponent.propDecorators = {
    logoType: [{ type: Input }],
    logoHeight: [{ type: Input }],
    logoColor: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    FxConfidentialityLogoComponent.prototype.confidentialityLogo;
    /** @type {?} */
    FxConfidentialityLogoComponent.prototype.logoType;
    /** @type {?} */
    FxConfidentialityLogoComponent.prototype.logoHeight;
    /** @type {?} */
    FxConfidentialityLogoComponent.prototype.logoColor;
}
