// @fx-lib/components/esm5/lib/fx-security/components/idp-login/idp-login.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component } from '@angular/core';
import { Router } from '@angular/router';
import { NotificationsService } from 'angular2-notifications';
import { NEVER, Subscription } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { AuthenticationService } from '../../services/authentication.service';
var IdpLoginComponent = /** @class */ (function () {
    function IdpLoginComponent(authenticationService, router, notificationService) {
        this.authenticationService = authenticationService;
        this.router = router;
        this.notificationService = notificationService;
        this.subscription = new Subscription();
    }
    /**
     * @return {?}
     */
    IdpLoginComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.subscription.add(this.authenticationService.loginFromUrlToken(window.location.href)
            .pipe(catchError((/**
         * @param {?} error
         * @return {?}
         */
        function (error) {
            _this.notificationService.error(error);
            return NEVER;
        })))
            .subscribe((/**
         * @param {?} identityToken
         * @return {?}
         */
        function (identityToken) {
            _this.router.navigateByUrl(identityToken.state);
        })));
    };
    /**
     * @return {?}
     */
    IdpLoginComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.subscription.unsubscribe();
    };
    IdpLoginComponent.decorators = [
        { type: Component, args: [{
                    template: ''
                }] }
    ];
    /** @nocollapse */
    IdpLoginComponent.ctorParameters = function () { return [
        { type: AuthenticationService },
        { type: Router },
        { type: NotificationsService }
    ]; };
    return IdpLoginComponent;
}());
export { IdpLoginComponent };
if (false) {
    /**
     * @type {?}
     * @private
     */
    IdpLoginComponent.prototype.subscription;
    /**
     * @type {?}
     * @private
     */
    IdpLoginComponent.prototype.authenticationService;
    /**
     * @type {?}
     * @private
     */
    IdpLoginComponent.prototype.router;
    /**
     * @type {?}
     * @private
     */
    IdpLoginComponent.prototype.notificationService;
}
