// @fx-lib/components/esm5/lib/fx-user-notification/services/fx-user-notification-local-development.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { of, timer } from 'rxjs';
var FxUserNotificationLocalDevelopmentService = /** @class */ (function () {
    function FxUserNotificationLocalDevelopmentService() {
        this.notifications = [];
        this.idCounterForMessages = 0;
        this.startTimer();
    }
    /**
     * @return {?}
     */
    FxUserNotificationLocalDevelopmentService.prototype.getNotifications = /**
     * @return {?}
     */
    function () {
        return of(this.notifications);
    };
    /**
     * @return {?}
     */
    FxUserNotificationLocalDevelopmentService.prototype.postUserReadLog = /**
     * @return {?}
     */
    function () {
        this.notifications.forEach((/**
         * @param {?} message
         * @return {?}
         */
        function (message) { return message.read = true; }));
        return of({});
    };
    /**
     * @private
     * @return {?}
     */
    FxUserNotificationLocalDevelopmentService.prototype.startTimer = /**
     * @private
     * @return {?}
     */
    function () {
        var _this = this;
        // this timer is on purpose not stopped or unsubscribed as it simulates a server that is generating messages,
        // it will not be stopped by the client.
        timer(0, 5000).subscribe((/**
         * @return {?}
         */
        function () {
            _this.notifications.push(_this.generateNewMessage());
        }));
    };
    /**
     * @private
     * @return {?}
     */
    FxUserNotificationLocalDevelopmentService.prototype.generateNewMessage = /**
     * @private
     * @return {?}
     */
    function () {
        this.idCounterForMessages++;
        /** @type {?} */
        var severity = this.getRandomSeverity();
        /** @type {?} */
        var messageText = Math.random() > 0.5
            ? "This is " + severity + " message " + this.idCounterForMessages.toString() + " locally generated."
            : "This is " + severity + " message " + this.idCounterForMessages.toString() + " locally generated. With a text which is longer than expected. This way a collapsible message can be demonstrated.";
        return {
            messageHeadline: "Dummy " + severity + " Message " + this.idCounterForMessages.toString(),
            updatedByUser: 'dummy',
            severity: severity,
            read: false,
            messageText: messageText,
            id: this.idCounterForMessages.toString(),
            audience: undefined,
            updateTimestamp: new Date()
        };
    };
    /**
     * @private
     * @return {?}
     */
    FxUserNotificationLocalDevelopmentService.prototype.getRandomSeverity = /**
     * @private
     * @return {?}
     */
    function () {
        /** @type {?} */
        var rnd = Math.random();
        if (rnd > 0.66) {
            return 'INFO';
        }
        else if (rnd > 0.33) {
            return 'WARNING';
        }
        else {
            return 'ERROR';
        }
    };
    FxUserNotificationLocalDevelopmentService.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    FxUserNotificationLocalDevelopmentService.ctorParameters = function () { return []; };
    return FxUserNotificationLocalDevelopmentService;
}());
export { FxUserNotificationLocalDevelopmentService };
if (false) {
    /** @type {?} */
    FxUserNotificationLocalDevelopmentService.prototype.notifications;
    /**
     * @type {?}
     * @private
     */
    FxUserNotificationLocalDevelopmentService.prototype.idCounterForMessages;
}
