// @fx-lib/components/esm5/lib/shared/shared.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { TranslatePipeMock } from './test/mocks/pipes/translate.pipe.mock';
import { TranslateWithPluralizationPipeMock } from './test/mocks/pipes/translate-with-pluralization.pipe.mock';
import { TranslateWithPluralizationPipe } from './pipes/translate-with-pluralization.pipe';
import { NgxUiLoaderModule } from 'ngx-ui-loader';
/** @type {?} */
var NGX_UI_LOADER_FX_CONFIG = {
    fgsColor: '#004666',
    fgsSize: 70,
    fgsType: 'circle',
    fgsPosition: 'center-center',
    overlayColor: 'rgba(255,255,255,0.9)',
    hasProgressBar: false,
    textColor: '#004666',
};
var SharedModule = /** @class */ (function () {
    function SharedModule() {
    }
    SharedModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        NgxUiLoaderModule.forRoot(NGX_UI_LOADER_FX_CONFIG),
                    ],
                    declarations: [
                        TranslatePipeMock,
                        TranslateWithPluralizationPipe,
                        TranslateWithPluralizationPipeMock,
                    ],
                    exports: [
                        TranslateWithPluralizationPipe,
                        NgxUiLoaderModule,
                    ]
                },] }
    ];
    return SharedModule;
}());
export { SharedModule };
