// @fx-lib/components/esm2015/lib/fx-boundary-selector/services/boundary-data.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable, Inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { AuthenticationService } from '../../fx-security/services/authentication.service';
import { SECURITY_CONFIG } from '../../fx-security/domain/security-config';
import { AuthorizationService } from '../../fx-security/services/authorization.service';
import { map } from 'rxjs/operators';
export class BoundaryDataService {
    /**
     * @param {?} http
     * @param {?} authenticationService
     * @param {?} authorizationService
     * @param {?} securityConfig
     */
    constructor(http, authenticationService, authorizationService, securityConfig) {
        this.http = http;
        this.authenticationService = authenticationService;
        this.authorizationService = authorizationService;
        this.securityConfig = securityConfig;
    }
    /**
     * @return {?}
     */
    getBoundariesForUserScope() {
        return this.http.get(this.getApiUrl()).pipe(map((/**
         * @param {?} boundarySelectorResources
         * @return {?}
         */
        (boundarySelectorResources) => {
            return this.filterBoundariesForUserScope(boundarySelectorResources);
        })));
    }
    /**
     * @param {?} boundarySelectorResources
     * @return {?}
     */
    filterBoundariesForUserScope(boundarySelectorResources) {
        /** @type {?} */
        const filteredBoundarySelectorResources = [];
        if (this.authorizationService.hasDevopsRole()) {
            return boundarySelectorResources;
        }
        this.authenticationService.getPayload().roles.forEach((/**
         * @param {?} role
         * @return {?}
         */
        (role) => {
            if (role.boundaries[this.securityConfig.boundary_name]) {
                role.boundaries[this.securityConfig.boundary_name].forEach((/**
                 * @param {?} scope
                 * @return {?}
                 */
                (scope) => {
                    /** @type {?} */
                    const boundaryFound = boundarySelectorResources.find((/**
                     * @param {?} boundary
                     * @return {?}
                     */
                    boundary => boundary.boundaryValueKey === scope));
                    if (boundaryFound) {
                        filteredBoundarySelectorResources.push(boundaryFound);
                    }
                }));
            }
        }));
        return filteredBoundarySelectorResources;
    }
    /**
     * @private
     * @return {?}
     */
    getApiUrl() {
        return `${this.securityConfig.authorization_service_url}/api/v1/domains/${this.securityConfig.app_domain}/apps/${this.securityConfig.app_name}/boundaryvaluesminified/`;
    }
}
BoundaryDataService.decorators = [
    { type: Injectable }
];
/** @nocollapse */
BoundaryDataService.ctorParameters = () => [
    { type: HttpClient },
    { type: AuthenticationService },
    { type: AuthorizationService },
    { type: undefined, decorators: [{ type: Inject, args: [SECURITY_CONFIG,] }] }
];
if (false) {
    /** @type {?} */
    BoundaryDataService.prototype.securityConfig;
    /**
     * @type {?}
     * @private
     */
    BoundaryDataService.prototype.http;
    /**
     * @type {?}
     * @private
     */
    BoundaryDataService.prototype.authenticationService;
    /**
     * @type {?}
     * @private
     */
    BoundaryDataService.prototype.authorizationService;
}
