// @fx-lib/components/esm2015/lib/fx-boundary-selector/services/boundary-selector-data.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
export class BoundarySelectorDataService {
    constructor() {
        this.boundaryResourceList$ = new BehaviorSubject([]);
        this.selectedBoundary$ = new BehaviorSubject(null);
    }
    /**
     * @param {?} boundaryResourceList
     * @return {?}
     */
    setBoundaryResourceList(boundaryResourceList) {
        if (this.boundaryResourceList$.getValue() === boundaryResourceList) {
            return;
        }
        this.boundaryResourceList$.next(boundaryResourceList);
        if (this.cachedBoundary != null) {
            this.setSelectedBoundaryByKey(this.cachedBoundary);
            this.cachedBoundary = null;
        }
        if (this.selectedBoundary$.getValue() == null) {
            this.setSelectedBoundaryToDefault();
        }
    }
    /**
     * @param {?} boundaryKey
     * @return {?}
     */
    setSelectedBoundaryByKey(boundaryKey) {
        /** @type {?} */
        const foundBoundary = this.getBoundaryByKey(boundaryKey);
        if (this.boundaryResourceList$.getValue().length < 1) {
            this.setSelectedBoundaryToDefault();
            this.cacheBoundaryKey(boundaryKey);
            return;
        }
        if (!foundBoundary) {
            this.setSelectedBoundaryToDefault();
            return;
        }
        if (this.selectedBoundary$.getValue() === foundBoundary) {
            return;
        }
        this.selectedBoundary$.next(foundBoundary);
    }
    /**
     * @param {?} boundary
     * @return {?}
     */
    setSelectedBoundary(boundary) {
        this.setSelectedBoundaryByKey(boundary.boundaryValueKey);
    }
    /**
     * @private
     * @param {?} boundaryKey
     * @return {?}
     */
    getBoundaryByKey(boundaryKey) {
        return this.boundaryResourceList$.getValue().find((/**
         * @param {?} x
         * @return {?}
         */
        x => x.boundaryValueKey === boundaryKey));
    }
    /**
     * @private
     * @return {?}
     */
    setSelectedBoundaryToDefault() {
        if (this.boundaryResourceList$.getValue() == null || this.boundaryResourceList$.getValue().length < 1) {
            this.selectedBoundary$.next(null);
            return;
        }
        this.selectedBoundary$.next(this.boundaryResourceList$.getValue()[0]);
    }
    /**
     * @private
     * @param {?} boundaryKey
     * @return {?}
     */
    cacheBoundaryKey(boundaryKey) {
        if (this.cachedBoundary === boundaryKey) {
            return;
        }
        this.cachedBoundary = boundaryKey;
    }
}
BoundarySelectorDataService.decorators = [
    { type: Injectable }
];
/** @nocollapse */
BoundarySelectorDataService.ctorParameters = () => [];
if (false) {
    /** @type {?} */
    BoundarySelectorDataService.prototype.boundaryResourceList$;
    /** @type {?} */
    BoundarySelectorDataService.prototype.selectedBoundary$;
    /**
     * @type {?}
     * @private
     */
    BoundarySelectorDataService.prototype.cachedBoundary;
}
