// @fx-lib/components/esm2015/lib/fx-security/services/authentication.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Inject, Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { BehaviorSubject } from 'rxjs';
import { map, tap } from 'rxjs/operators';
import { getBase64DecodeUrlSafe, getBase64EncodeUrlSafe } from '../../shared/helper/base64.helper';
import { SECURITY_CONFIG } from '../domain/security-config';
import { LoginService } from './login.service';
import { RedirectorService } from './redirector.service';
/**
 * @record
 */
function HashParams() { }
/** @type {?} */
export const TOKEN_KEY = 'URS_TOKEN';
/** @type {?} */
export const ID_TOKEN_KEY = 'URS_ID_TOKEN';
/**
 * @record
 */
export function IdpIdentityToken() { }
if (false) {
    /** @type {?} */
    IdpIdentityToken.prototype.token;
    /** @type {?} */
    IdpIdentityToken.prototype.state;
    /** @type {?} */
    IdpIdentityToken.prototype.app;
}
export class AuthenticationService {
    /**
     * @param {?} loginService
     * @param {?} router
     * @param {?} redirectorService
     * @param {?} securityConfig
     */
    constructor(loginService, router, redirectorService, securityConfig) {
        this.loginService = loginService;
        this.router = router;
        this.redirectorService = redirectorService;
        this.isLoggedIn = false;
        this.loginEvents = new BehaviorSubject('loggedout');
        this.initTokenFromStorage();
        this.securityConfig = securityConfig;
    }
    /**
     * @return {?}
     */
    get token() {
        return this._token || '';
    }
    /**
     * @param {?} token
     * @return {?}
     */
    set token(token) {
        this._token = token;
    }
    /**
     * @param {?} identityToken
     * @return {?}
     */
    login(identityToken) {
        return this.loginService.login(identityToken)
            .pipe(tap((/**
         * @param {?} token
         * @return {?}
         */
        (token) => {
            this.initLogin(token, identityToken);
            this.loginEvents.next('loggedin');
        }), (/**
         * @return {?}
         */
        () => {
            this.removeLoginInformation();
            this.router.navigateByUrl('/login-error');
        })), map((/**
         * @return {?}
         */
        () => identityToken)));
    }
    /**
     * @return {?}
     */
    logout() {
        // Check for ID token
        /** @type {?} */
        const idToken = localStorage.getItem(ID_TOKEN_KEY);
        this.removeLoginInformation();
        if (idToken) {
            this.redirectorService.redirectToUrl(this.buildIdpLogoutUrl(idToken));
        }
        else {
            this.router.navigateByUrl('/');
        }
    }
    /**
     * @return {?}
     */
    getAuthTokenHeaderString() {
        return 'Bearer ' + this.token;
    }
    /**
     * @return {?}
     */
    initTokenFromStorage() {
        /** @type {?} */
        const storedToken = localStorage.getItem(TOKEN_KEY);
        if (storedToken == null) {
            return;
        }
        try {
            /** @type {?} */
            const parts = storedToken.split('.');
            /** @type {?} */
            const payload = JSON.parse(atob(parts[1]));
            /** @type {?} */
            const until = new Date(payload.exp * 1000);
            /** @type {?} */
            const now = new Date();
            if (now < until) {
                this.token = storedToken;
                this.isLoggedIn = true;
                this.loginEvents.next('loggedin');
            }
        }
        catch (ex) {
            this.removeLoginInformation();
        }
    }
    /**
     * @param {?} url
     * @return {?}
     */
    loginFromUrlToken(url) {
        /** @type {?} */
        const identityToken = this.extractTokenFromUrl(url);
        return this.login(identityToken);
    }
    /**
     * @return {?}
     */
    getPayload() {
        if (this.isLoggedIn) {
            /** @type {?} */
            const token = this.token;
            /** @type {?} */
            const parts = token.split('.');
            if (parts.length >= 2) {
                return JSON.parse(atob(parts[1]));
            }
        }
        return { roles: [], username: '' };
    }
    /**
     * @return {?}
     */
    loginWithRedirectUrl() {
        this.redirectToIdpLogin(this.securityConfig.login_redirect_url);
    }
    /**
     * @param {?} targetUrl
     * @return {?}
     */
    redirectToIdpLogin(targetUrl) {
        this.redirectorService.redirectToUrl(this.buildIdpLoginUrl(targetUrl));
    }
    /**
     * @private
     * @param {?} token
     * @param {?} identityToken
     * @return {?}
     */
    initLogin(token, identityToken) {
        this.token = token;
        this.isLoggedIn = true;
        localStorage.setItem(TOKEN_KEY, token);
        localStorage.setItem(ID_TOKEN_KEY, identityToken.token);
    }
    /**
     * @private
     * @return {?}
     */
    removeLoginInformation() {
        this.token = undefined;
        this.isLoggedIn = false;
        localStorage.removeItem(TOKEN_KEY);
        localStorage.removeItem(ID_TOKEN_KEY);
        this.loginEvents.next('loggedout');
    }
    /**
     * @private
     * @param {?} state
     * @return {?}
     */
    buildIdpLoginUrl(state) {
        return this.securityConfig.idp_base_url + 'auth'
            + '?response_type=id_token'
            + '&scope=openid'
            + '&nonce=' + this.generateNonce()
            + '&client_id=' + encodeURIComponent(this.securityConfig.client_id)
            + '&redirect_uri=' + encodeURIComponent(this.securityConfig.login_redirect_url)
            + '&state=' + encodeURIComponent(getBase64EncodeUrlSafe(state));
    }
    /**
     * @private
     * @return {?}
     */
    mathFloorRandom() {
        return Math.floor((1 + Math.random()) * 0x10000)
            .toString(16)
            .substring(1);
    }
    /**
     * @private
     * @return {?}
     */
    generateNonce() {
        /** @type {?} */
        let nonce = '';
        for (let i = 0; i < 8; i++) {
            nonce += this.mathFloorRandom();
        }
        return nonce;
    }
    /**
     * @private
     * @param {?} url
     * @return {?}
     */
    extractTokenFromUrl(url) {
        /** @type {?} */
        const hashPart = this.extractHashPart(url);
        /** @type {?} */
        const hashParams = this.extractParameters(decodeURIComponent(hashPart));
        if (hashParams && hashParams.state && hashParams.id_token) {
            return {
                state: hashParams.state,
                token: hashParams.id_token,
                app: this.securityConfig.app_name,
            };
        }
        return null;
    }
    /**
     * @private
     * @param {?} url
     * @return {?}
     */
    extractHashPart(url) {
        /** @type {?} */
        const hashPosition = url.indexOf('#');
        if (hashPosition >= 0 && url.length > hashPosition) {
            return url.slice(hashPosition + 1);
        }
        else {
            return '';
        }
    }
    /**
     * @private
     * @param {?} paramString
     * @return {?}
     */
    extractParameters(paramString) {
        return paramString.split('&')
            .reduce((/**
         * @param {?} params
         * @param {?} keyValueText
         * @return {?}
         */
        (params, keyValueText) => {
            if (keyValueText) {
                /** @type {?} */
                const keyValue = keyValueText.split('=');
                if (keyValue.length === 2) {
                    /** @type {?} */
                    const key = keyValue[0];
                    /** @type {?} */
                    const value = keyValue[1];
                    /** @type {?} */
                    const decodedValue = key === 'state' ? getBase64DecodeUrlSafe(value) : value;
                    params[key] = decodeURIComponent(decodedValue);
                }
            }
            return params;
        }), {});
    }
    /**
     * @private
     * @param {?} token
     * @return {?}
     */
    buildIdpLogoutUrl(token) {
        return this.securityConfig.idp_base_url + 'logout'
            + '?id_token=' + token
            + '&post_logout_redirect_uri=' + encodeURIComponent(this.securityConfig.logout_redirect_url);
    }
}
AuthenticationService.decorators = [
    { type: Injectable }
];
/** @nocollapse */
AuthenticationService.ctorParameters = () => [
    { type: LoginService },
    { type: Router },
    { type: RedirectorService },
    { type: undefined, decorators: [{ type: Inject, args: [SECURITY_CONFIG,] }] }
];
if (false) {
    /**
     * @type {?}
     * @protected
     */
    AuthenticationService.prototype.securityConfig;
    /** @type {?} */
    AuthenticationService.prototype.isLoggedIn;
    /**
     * @type {?}
     * @protected
     */
    AuthenticationService.prototype._token;
    /** @type {?} */
    AuthenticationService.prototype.loginEvents;
    /**
     * @type {?}
     * @private
     */
    AuthenticationService.prototype.loginService;
    /**
     * @type {?}
     * @private
     */
    AuthenticationService.prototype.router;
    /**
     * @type {?}
     * @private
     */
    AuthenticationService.prototype.redirectorService;
}
