// @fx-lib/components/esm2015/lib/fx-security/services/authorization.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable, Inject } from '@angular/core';
import { AuthenticationService } from './authentication.service';
import { SECURITY_CONFIG } from '../domain/security-config';
export class AuthorizationService {
    /**
     * @param {?} authenticationService
     * @param {?} securityConfig
     */
    constructor(authenticationService, securityConfig) {
        this.authenticationService = authenticationService;
        this.securityConfig = securityConfig;
    }
    /**
     * @return {?}
     */
    hasViewRole() {
        return this.checkMultipleRoles(this.securityConfig.roles.view);
    }
    /**
     * @param {?} boundaryValue
     * @return {?}
     */
    hasViewRoleForBoundary(boundaryValue) {
        return this.checkMultipleRolesForBoundary(this.securityConfig.roles.view, boundaryValue);
    }
    /**
     * @return {?}
     */
    hasEditRole() {
        return this.checkMultipleRoles(this.securityConfig.roles.edit);
    }
    /**
     * @param {?} boundaryValue
     * @return {?}
     */
    hasEditRoleForBoundary(boundaryValue) {
        return this.checkMultipleRolesForBoundary(this.securityConfig.roles.edit, boundaryValue);
    }
    /**
     * @return {?}
     */
    hasDevopsRole() {
        return this.checkMultipleRoles(this.securityConfig.roles.devops);
    }
    /**
     * @param {?} roleToCheck
     * @return {?}
     */
    hasRole(roleToCheck) {
        /** @type {?} */
        const payload = this.authenticationService.getPayload();
        return payload.roles.some((/**
         * @param {?} role
         * @return {?}
         */
        (role) => role.role === roleToCheck));
    }
    /**
     * @param {?} roleToCheck
     * @param {?} boundaryValue
     * @return {?}
     */
    hasRoleForBoundary(roleToCheck, boundaryValue) {
        /** @type {?} */
        const payload = this.authenticationService.getPayload();
        /** @type {?} */
        const foundRole = payload.roles.find((/**
         * @param {?} r
         * @return {?}
         */
        r => r.role === roleToCheck));
        if (!foundRole) {
            return false;
        }
        /** @type {?} */
        const devopsRole = this.securityConfig.roles.devops.find((/**
         * @param {?} r
         * @return {?}
         */
        r => r === foundRole.role));
        if (devopsRole) {
            return true;
        }
        /** @type {?} */
        const boundaries = foundRole.boundaries[this.securityConfig.boundary_name];
        return boundaries.some((/**
         * @param {?} boundary
         * @return {?}
         */
        (boundary) => boundary === boundaryValue));
    }
    /**
     * @private
     * @param {?} roles
     * @return {?}
     */
    checkMultipleRoles(roles) {
        return roles.some((/**
         * @param {?} role
         * @return {?}
         */
        role => this.hasRole(role)));
    }
    /**
     * @private
     * @param {?} roles
     * @param {?} boundaryValue
     * @return {?}
     */
    checkMultipleRolesForBoundary(roles, boundaryValue) {
        return roles.some((/**
         * @param {?} role
         * @return {?}
         */
        role => this.hasRoleForBoundary(role, boundaryValue)));
    }
}
AuthorizationService.decorators = [
    { type: Injectable }
];
/** @nocollapse */
AuthorizationService.ctorParameters = () => [
    { type: AuthenticationService },
    { type: undefined, decorators: [{ type: Inject, args: [SECURITY_CONFIG,] }] }
];
if (false) {
    /** @type {?} */
    AuthorizationService.prototype.securityConfig;
    /**
     * @type {?}
     * @private
     */
    AuthorizationService.prototype.authenticationService;
}
