// @fx-lib/components/esm2015/lib/fx-security/services/login.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Inject, Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { throwError } from 'rxjs';
import { catchError, map } from 'rxjs/operators';
import { SECURITY_CONFIG } from '../domain/security-config';
export class LoginService {
    /**
     * @param {?} http
     * @param {?} securityConfig
     */
    constructor(http, securityConfig) {
        this.http = http;
        this.ursUrl = 'api/v1/auth';
        this.securityConfig = securityConfig;
    }
    /**
     * @param {?} idToken
     * @return {?}
     */
    login(idToken) {
        /** @type {?} */
        const url = this.securityConfig.authorization_service_url + this.ursUrl;
        return this.http
            .post(url, idToken, { observe: 'response', responseType: 'text' })
            .pipe(map((/**
         * @param {?} response
         * @return {?}
         */
        (response) => response.body)), catchError((/**
         * @param {?} error
         * @return {?}
         */
        (error) => this.handleError(error))));
    }
    /**
     * @private
     * @param {?} error
     * @return {?}
     */
    handleError(error) {
        /** @type {?} */
        const errorMessage = error.message ? error.message : error.toString();
        console.error(errorMessage);
        return throwError(errorMessage);
    }
}
LoginService.decorators = [
    { type: Injectable }
];
/** @nocollapse */
LoginService.ctorParameters = () => [
    { type: HttpClient },
    { type: undefined, decorators: [{ type: Inject, args: [SECURITY_CONFIG,] }] }
];
if (false) {
    /** @type {?} */
    LoginService.prototype.securityConfig;
    /**
     * @type {?}
     * @private
     */
    LoginService.prototype.ursUrl;
    /**
     * @type {?}
     * @private
     */
    LoginService.prototype.http;
}
