// @fx-lib/components/esm5/lib/fx-header/fx-header.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FxHeaderComponent } from './fx-header.component';
import { FxBoundarySelectorModule } from '../fx-boundary-selector/fx-boundary-selector.module';
import { FxSecurityModule } from '../fx-security/fx-security.module';
import { FxConfidentialityLogoModule } from '../fx-confidentiality-logo/fx-confidentiality-logo.module';
import { FxBoundarySelectorComponent } from '../fx-boundary-selector/fx-boundary-selector.component';
import { FxUserSelectorComponent } from '../fx-security/components/fx-user-selector/fx-user-selector.component';
import { FxWikiModule } from '../fx-wiki/fx-wiki.module';
import { FxHeaderConfigValidatorService } from './services/fx-header-config-validator.service';
import { FxHeaderConfigService } from './services/fx-header-config.service';
import { FxLoginComponent } from '../fx-security/components/fx-login/fx-login.component';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { FxTranslationModule } from '../fx-translation/fx-translation.module';
import { RouterModule } from '@angular/router';
import { FxHeaderAppSpecificTextComponent } from './components/fx-header-app-specific-text.component';
import { FxWikiComponent } from '../fx-wiki/fx-wiki.component';
import { FxConfidentialityLogoComponent } from '../fx-confidentiality-logo/fx-confidentiality-logo.component';
var FxHeaderModule = /** @class */ (function () {
    function FxHeaderModule() {
    }
    FxHeaderModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [
                        FxHeaderComponent,
                        FxHeaderAppSpecificTextComponent
                    ],
                    exports: [
                        FxHeaderComponent,
                        FxHeaderAppSpecificTextComponent
                    ],
                    imports: [
                        RouterModule,
                        CommonModule,
                        FxBoundarySelectorModule,
                        FxSecurityModule,
                        FxConfidentialityLogoModule,
                        FxWikiModule,
                        NgbModule,
                        FxTranslationModule,
                    ],
                    providers: [
                        FxHeaderConfigService,
                        FxHeaderConfigValidatorService
                    ],
                    entryComponents: [FxBoundarySelectorComponent, FxUserSelectorComponent, FxLoginComponent, FxWikiComponent, FxConfidentialityLogoComponent, FxHeaderAppSpecificTextComponent]
                },] }
    ];
    return FxHeaderModule;
}());
export { FxHeaderModule };
