// @fx-lib/components/esm5/lib/fx-custom-table/fx-custom-table.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { BrowserModule } from '@angular/platform-browser';
import { RouterModule } from '@angular/router';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { DataTableBodyComponent, NgxDatatableModule } from '@swimlane/ngx-datatable';
import { ContextMenuModule } from 'ngx-contextmenu';
import { FxTranslationModule } from '../fx-translation/fx-translation.module';
import { SharedModule } from '../shared/shared.module';
import { FxCustomTableComponent } from './fx-custom-table.component';
import { FxColorIndicatorsCellTemplateComponent } from './templates/color-indicators-cell-template/fx-color-indicators-cell-template.component';
/**
 * nasty fix for:
 * DataTableBodyComponent.html:64 ERROR TypeError: Cannot read property '0' of undefined at Object.eval [as updateDirectives] (DataTableBodyComponent.html:37)
 */
DataTableBodyComponent.prototype.updateRows = (/**
 * @return {?}
 */
function () {
    var _a = this.indexes, first = _a.first, last = _a.last;
    /** @type {?} */
    var rowIndex = first;
    /** @type {?} */
    var idx = 0;
    /** @type {?} */
    var temp = [];
    this.rowIndexes.clear();
    if (this.groupedRows) {
        while (rowIndex < last && rowIndex < this.groupedRows.length) {
            // Add the groups into this page
            /** @type {?} */
            var group = this.groupedRows[rowIndex];
            temp[idx] = group;
            idx++;
            // Group index in this context
            rowIndex++;
        }
    }
    else {
        while (rowIndex < last && rowIndex < this.rowCount) {
            /** @type {?} */
            var row = this.rows[rowIndex];
            if (row) {
                this.rowIndexes.set(row, rowIndex);
                temp[idx] = row;
            }
            idx++;
            rowIndex++;
        }
    }
    this.temp = temp.filter((/**
     * @param {?} x
     * @return {?}
     */
    function (x) { return !!x; }));
});
var FxCustomTableModule = /** @class */ (function () {
    function FxCustomTableModule() {
    }
    FxCustomTableModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [
                        FxColorIndicatorsCellTemplateComponent,
                        FxCustomTableComponent,
                    ],
                    imports: [
                        BrowserModule,
                        ContextMenuModule.forRoot(),
                        FormsModule,
                        FxTranslationModule,
                        NgbModule,
                        NgxDatatableModule,
                        SharedModule,
                        RouterModule,
                    ],
                    exports: [
                        FxColorIndicatorsCellTemplateComponent,
                        FxCustomTableComponent,
                    ],
                },] }
    ];
    return FxCustomTableModule;
}());
export { FxCustomTableModule };
