// @fx-lib/components/esm2015/lib/fx-user-notification/services/fx-user-notification-local-development.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { of, timer } from 'rxjs';
/**
 * A implementation of IFxUserNotificationLocalDevelopmentService just without server, it generates all data offline.
 * <b>Purpose:</b>
 * Ease of development and demonstration in a showcase.
 */
export class FxUserNotificationLocalDevelopmentService {
    constructor() {
        this.notifications = [];
        this.idCounterForMessages = 0;
        this.startTimer();
    }
    /**
     * @return {?}
     */
    getNotifications() {
        return of(this.notifications);
    }
    /**
     * @return {?}
     */
    postUserReadLog() {
        this.notifications.forEach((/**
         * @param {?} message
         * @return {?}
         */
        (message) => message.read = true));
        return of({});
    }
    /**
     * @private
     * @return {?}
     */
    startTimer() {
        // this timer is on purpose not stopped or unsubscribed as it simulates a server that is generating messages,
        // it will not be stopped by the client.
        timer(0, 5000).subscribe((/**
         * @return {?}
         */
        () => {
            this.notifications.push(this.generateNewMessage());
        }));
    }
    /**
     * @private
     * @return {?}
     */
    generateNewMessage() {
        this.idCounterForMessages++;
        /** @type {?} */
        const severity = this.getRandomSeverity();
        /** @type {?} */
        const messageText = Math.random() > 0.5
            ? `This is ${severity} message ${this.idCounterForMessages.toString()} locally generated.`
            : `This is ${severity} message ${this.idCounterForMessages.toString()} locally generated. With a text which is longer than expected. This way a collapsible message can be demonstrated.`;
        return {
            messageHeadline: `Dummy ${severity} Message ${this.idCounterForMessages.toString()}`,
            updatedByUser: 'dummy',
            severity: severity,
            read: false,
            messageText: messageText,
            id: this.idCounterForMessages.toString(),
            audience: undefined,
            updateTimestamp: new Date()
        };
    }
    /**
     * @private
     * @return {?}
     */
    getRandomSeverity() {
        /** @type {?} */
        const rnd = Math.random();
        if (rnd > 0.66) {
            return 'INFO';
        }
        else if (rnd > 0.33) {
            return 'WARNING';
        }
        else {
            return 'ERROR';
        }
    }
}
FxUserNotificationLocalDevelopmentService.decorators = [
    { type: Injectable }
];
/** @nocollapse */
FxUserNotificationLocalDevelopmentService.ctorParameters = () => [];
if (false) {
    /** @type {?} */
    FxUserNotificationLocalDevelopmentService.prototype.notifications;
    /**
     * @type {?}
     * @private
     */
    FxUserNotificationLocalDevelopmentService.prototype.idCounterForMessages;
}
