// @fx-lib/components/esm2015/lib/fx-security/directives/has-role.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { ChangeDetectorRef, Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
import { AuthenticationService } from '../services/authentication.service';
import { AuthorizationService } from '../services/authorization.service';
export class HasRoleDirective {
    /**
     * @param {?} _viewContainer
     * @param {?} templateRef
     * @param {?} authService
     * @param {?} authorizationService
     * @param {?} changeDetector
     */
    constructor(_viewContainer, templateRef, authService, authorizationService, changeDetector) {
        this._viewContainer = _viewContainer;
        this.authorizationService = authorizationService;
        this.changeDetector = changeDetector;
        this._context = new HasRoleContext();
        this._thenTemplateRef = null;
        this._elseTemplateRef = null;
        this._thenViewRef = null;
        this._elseViewRef = null;
        this._thenTemplateRef = templateRef;
        this.loginEventsSubscription = authService.loginEvents.subscribe((/**
         * @return {?}
         */
        () => {
            this._updateView();
            this.changeDetector.markForCheck();
        }));
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        if (this.loginEventsSubscription) {
            this.loginEventsSubscription.unsubscribe();
        }
    }
    /**
     * @param {?} condition
     * @return {?}
     */
    set hasRole(condition) {
        this._context.$implicit = this._context.hasRole = condition;
        this._updateView();
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set hasRoleFactory(value) {
        this._context.factory = value;
        this._updateView();
    }
    /**
     * @param {?} templateRef
     * @return {?}
     */
    set hasRoleThen(templateRef) {
        this._thenTemplateRef = templateRef;
        this._thenViewRef = null; // clear previous view if any.
        this._updateView();
    }
    /**
     * @param {?} templateRef
     * @return {?}
     */
    set hasRoleElse(templateRef) {
        this._elseTemplateRef = templateRef;
        this._elseViewRef = null; // clear previous view if any.
        this._updateView();
    }
    /**
     * @private
     * @return {?}
     */
    _updateView() {
        /** @type {?} */
        let hasRole = false;
        if (this._context && this._context.$implicit) {
            if (typeof this._context.factory === 'string' && this._context.factory.length > 0) {
                hasRole = this.authorizationService.hasRoleForBoundary(this._context.$implicit, this._context.factory);
            }
            else {
                hasRole = this._context && this._context.$implicit && this.authorizationService.hasRole(this._context.$implicit);
            }
        }
        if (hasRole) {
            if (!this._thenViewRef) {
                this._viewContainer.clear();
                this._elseViewRef = null;
                if (this._thenTemplateRef) {
                    this._thenViewRef =
                        this._viewContainer.createEmbeddedView(this._thenTemplateRef, this._context);
                }
            }
        }
        else {
            if (!this._elseViewRef) {
                this._viewContainer.clear();
                this._thenViewRef = null;
                if (this._elseTemplateRef) {
                    this._elseViewRef =
                        this._viewContainer.createEmbeddedView(this._elseTemplateRef, this._context);
                }
            }
        }
    }
}
HasRoleDirective.decorators = [
    { type: Directive, args: [{
                selector: '[appHasRole]'
            },] }
];
/** @nocollapse */
HasRoleDirective.ctorParameters = () => [
    { type: ViewContainerRef },
    { type: TemplateRef },
    { type: AuthenticationService },
    { type: AuthorizationService },
    { type: ChangeDetectorRef }
];
HasRoleDirective.propDecorators = {
    hasRole: [{ type: Input }],
    hasRoleFactory: [{ type: Input }],
    hasRoleThen: [{ type: Input }],
    hasRoleElse: [{ type: Input }]
};
if (false) {
    /**
     * @type {?}
     * @private
     */
    HasRoleDirective.prototype._context;
    /**
     * @type {?}
     * @private
     */
    HasRoleDirective.prototype._thenTemplateRef;
    /**
     * @type {?}
     * @private
     */
    HasRoleDirective.prototype._elseTemplateRef;
    /**
     * @type {?}
     * @private
     */
    HasRoleDirective.prototype._thenViewRef;
    /**
     * @type {?}
     * @private
     */
    HasRoleDirective.prototype._elseViewRef;
    /**
     * @type {?}
     * @private
     */
    HasRoleDirective.prototype.loginEventsSubscription;
    /**
     * @type {?}
     * @private
     */
    HasRoleDirective.prototype._viewContainer;
    /**
     * @type {?}
     * @private
     */
    HasRoleDirective.prototype.authorizationService;
    /**
     * @type {?}
     * @private
     */
    HasRoleDirective.prototype.changeDetector;
}
export class HasRoleContext {
    constructor() {
        this.$implicit = null;
        this.hasRole = null;
        this.factory = null;
    }
}
if (false) {
    /** @type {?} */
    HasRoleContext.prototype.$implicit;
    /** @type {?} */
    HasRoleContext.prototype.hasRole;
    /** @type {?} */
    HasRoleContext.prototype.factory;
}
