// @fx-lib/components/esm2015/lib/fx-security/directives/is-logged-in.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { ChangeDetectorRef, Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
import { AuthenticationService } from '../services/authentication.service';
export class IsLoggedInDirective {
    /**
     * @param {?} authService
     * @param {?} _viewContainer
     * @param {?} templateRef
     * @param {?} changeDetector
     */
    constructor(authService, _viewContainer, templateRef, changeDetector) {
        this.authService = authService;
        this._viewContainer = _viewContainer;
        this.changeDetector = changeDetector;
        this._context = new IsLoggedInContext();
        this._thenTemplateRef = null;
        this._elseTemplateRef = null;
        this._thenViewRef = null;
        this._elseViewRef = null;
        this._thenTemplateRef = templateRef;
        this.loginEventsSubscription = authService.loginEvents.subscribe((/**
         * @return {?}
         */
        () => {
            this._updateView();
            this.changeDetector.markForCheck();
        }));
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        if (this.loginEventsSubscription) {
            this.loginEventsSubscription.unsubscribe();
        }
    }
    /**
     * @param {?} condition
     * @return {?}
     */
    set hasRole(condition) {
        this._context.$implicit = condition;
        this._updateView();
    }
    /**
     * @param {?} templateRef
     * @return {?}
     */
    set hasRoleThen(templateRef) {
        this._thenTemplateRef = templateRef;
        this._thenViewRef = null; // clear previous view if any.
        this._updateView();
    }
    /**
     * @param {?} templateRef
     * @return {?}
     */
    set hasRoleElse(templateRef) {
        this._elseTemplateRef = templateRef;
        this._elseViewRef = null; // clear previous view if any.
        this._updateView();
    }
    /**
     * @private
     * @return {?}
     */
    _updateView() {
        if (this.authService.isLoggedIn) {
            if (!this._thenViewRef) {
                this._viewContainer.clear();
                this._elseViewRef = null;
                if (this._thenTemplateRef) {
                    this._thenViewRef =
                        this._viewContainer.createEmbeddedView(this._thenTemplateRef, this._context);
                }
            }
        }
        else {
            if (!this._elseViewRef) {
                this._viewContainer.clear();
                this._thenViewRef = null;
                if (this._elseTemplateRef) {
                    this._elseViewRef =
                        this._viewContainer.createEmbeddedView(this._elseTemplateRef, this._context);
                }
            }
        }
    }
}
IsLoggedInDirective.decorators = [
    { type: Directive, args: [{
                selector: '[appIsLoggedIn]'
            },] }
];
/** @nocollapse */
IsLoggedInDirective.ctorParameters = () => [
    { type: AuthenticationService },
    { type: ViewContainerRef },
    { type: TemplateRef },
    { type: ChangeDetectorRef }
];
IsLoggedInDirective.propDecorators = {
    hasRole: [{ type: Input }],
    hasRoleThen: [{ type: Input }],
    hasRoleElse: [{ type: Input }]
};
if (false) {
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype._context;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype._thenTemplateRef;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype._elseTemplateRef;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype._thenViewRef;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype._elseViewRef;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype.loginEventsSubscription;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype.authService;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype._viewContainer;
    /**
     * @type {?}
     * @private
     */
    IsLoggedInDirective.prototype.changeDetector;
}
export class IsLoggedInContext {
    constructor() {
        this.$implicit = null;
    }
}
if (false) {
    /** @type {?} */
    IsLoggedInContext.prototype.$implicit;
}
