// @fx-lib/components/esm5/lib/fx-security/guards/auth-guard.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { AuthenticationService } from '../services/authentication.service';
import { AuthorizationService } from '../services/authorization.service';
/** @type {?} */
var NO_PERMISSION_ROUTE = 'no-permission';
var AuthGuard = /** @class */ (function () {
    function AuthGuard(authenticationService, authorizationService, router) {
        this.authenticationService = authenticationService;
        this.authorizationService = authorizationService;
        this.router = router;
    }
    /**
     * @param {?} route
     * @param {?} state
     * @return {?}
     */
    AuthGuard.prototype.canActivate = /**
     * @param {?} route
     * @param {?} state
     * @return {?}
     */
    function (route, state) {
        return this.checkLogin(state.url) && this.checkRoles(route.data) && this.checkBoundaryValue(route.data, route.paramMap);
    };
    /**
     * @private
     * @param {?} url
     * @return {?}
     */
    AuthGuard.prototype.checkLogin = /**
     * @private
     * @param {?} url
     * @return {?}
     */
    function (url) {
        if (this.authenticationService.isLoggedIn) {
            return true;
        }
        this.authenticationService.redirectToIdpLogin(url);
        return false;
    };
    /**
     * @private
     * @param {?} data
     * @param {?} paramMap
     * @return {?}
     */
    AuthGuard.prototype.checkBoundaryValue = /**
     * @private
     * @param {?} data
     * @param {?} paramMap
     * @return {?}
     */
    function (data, paramMap) {
        var _this = this;
        /** @type {?} */
        var hasPermission = false;
        if (!data || !data.boundaryParamName) {
            return true;
        }
        if (this.authorizationService.hasDevopsRole()) {
            return true;
        }
        this.authenticationService.getPayload().roles.forEach((/**
         * @param {?} role
         * @return {?}
         */
        function (role) {
            if (_this.authorizationService.hasRoleForBoundary(role.role, paramMap.get(data.boundaryParamName))) {
                hasPermission = true;
                return;
            }
        }));
        if (!hasPermission) {
            this.router.navigateByUrl(NO_PERMISSION_ROUTE);
        }
        return hasPermission;
    };
    /**
     * @private
     * @param {?} data
     * @return {?}
     */
    AuthGuard.prototype.checkRoles = /**
     * @private
     * @param {?} data
     * @return {?}
     */
    function (data) {
        var _this = this;
        /** @type {?} */
        var hasPermission = false;
        if (!data || !data.roles) {
            return true;
        }
        data.roles.forEach((/**
         * @param {?} role
         * @return {?}
         */
        function (role) {
            /** @type {?} */
            var requiredRoleName = role;
            /** @type {?} */
            var hasRole = _this.authorizationService.hasRole(requiredRoleName);
            if (hasRole) {
                hasPermission = true;
                return;
            }
        }));
        if (!hasPermission) {
            this.router.navigateByUrl(NO_PERMISSION_ROUTE);
        }
        return hasPermission;
    };
    AuthGuard.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    AuthGuard.ctorParameters = function () { return [
        { type: AuthenticationService },
        { type: AuthorizationService },
        { type: Router }
    ]; };
    return AuthGuard;
}());
export { AuthGuard };
if (false) {
    /**
     * @type {?}
     * @private
     */
    AuthGuard.prototype.authenticationService;
    /**
     * @type {?}
     * @private
     */
    AuthGuard.prototype.authorizationService;
    /**
     * @type {?}
     * @private
     */
    AuthGuard.prototype.router;
}
