// @fx-lib/components/esm2015/lib/fx-notification/fx-notification.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { ChangeDetectionStrategy, Component } from '@angular/core';
export class FxNotificationComponent {
    constructor() {
        this.notificationOptions = {
            position: ['top', 'right'],
            timeOut: 50000,
            clickToClose: true,
            showProgressBar: true,
            pauseOnHover: true,
            preventDuplicates: true,
        };
    }
}
FxNotificationComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-fx-notification',
                template: "<simple-notifications [options]=\"notificationOptions\"></simple-notifications>\r\n",
                changeDetection: ChangeDetectionStrategy.OnPush,
                styles: [""]
            }] }
];
/** @nocollapse */
FxNotificationComponent.ctorParameters = () => [];
if (false) {
    /** @type {?} */
    FxNotificationComponent.prototype.notificationOptions;
}
