// @fx-lib/components/esm5/lib/fx-user-notification/components/fx-user-notification/fx-user-notification.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { faBell } from '@fortawesome/free-solid-svg-icons';
import { FxUserNotificationService } from '../../services/fx-user-notification.service';
import { priorityMap } from '../../domain/fx-user-notification';
import { timer } from 'rxjs';
var FxUserNotificationComponent = /** @class */ (function () {
    function FxUserNotificationComponent(userNotificationService) {
        this.userNotificationService = userNotificationService;
        this.opened = false;
        this.icon = faBell;
        this.color = 'white';
        this.size = '2x';
        this.refreshInterval = 60000;
        this.charsOfShortText = 50;
        this.userNotificationList = [];
        this.userNotificationUnreadList = [];
        this.userNotificationReadList = [];
    }
    /**
     * @return {?}
     */
    FxUserNotificationComponent.prototype.getNotificationIconColor = /**
     * @return {?}
     */
    function () {
        return { color: this.color };
    };
    /**
     * @return {?}
     */
    FxUserNotificationComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.intervalTimeSubscription = timer(0, this.refreshInterval).subscribe((/**
         * @return {?}
         */
        function () {
            _this.receiveNotifications();
        }));
    };
    /**
     * @return {?}
     */
    FxUserNotificationComponent.prototype.receiveNotifications = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.userNotificationSubscription = this.userNotificationService.getNotifications().subscribe((/**
         * @param {?} res
         * @return {?}
         */
        function (res) {
            _this.sortNotificationsAndRemember(res);
        }));
    };
    /**
     * @return {?}
     */
    FxUserNotificationComponent.prototype.toggleNotificationArea = /**
     * @return {?}
     */
    function () {
        this.opened = !this.opened;
        if (!this.opened) {
            this.postReadNotificationLogAndRefreshNotifications();
        }
    };
    /**
     * @param {?} notificationData
     * @return {?}
     */
    FxUserNotificationComponent.prototype.sortNotificationsAndRemember = /**
     * @param {?} notificationData
     * @return {?}
     */
    function (notificationData) {
        this.userNotificationReadList = notificationData.filter((/**
         * @param {?} notification
         * @return {?}
         */
        function (notification) {
            return notification.read;
        }));
        this.userNotificationUnreadList = notificationData.filter((/**
         * @param {?} notification
         * @return {?}
         */
        function (notification) {
            return !notification.read;
        }));
        this.userNotificationList = this.sortNotification(this.userNotificationUnreadList).concat(this.sortNotification(this.userNotificationReadList));
        this.notificationCount = this.userNotificationUnreadList.length;
    };
    /**
     * @param {?} notificationData
     * @return {?}
     */
    FxUserNotificationComponent.prototype.sortNotification = /**
     * @param {?} notificationData
     * @return {?}
     */
    function (notificationData) {
        var _this = this;
        return notificationData.sort((/**
         * @param {?} notification1
         * @param {?} notification2
         * @return {?}
         */
        function (notification1, notification2) {
            {
                if (priorityMap.get(notification1.severity) < priorityMap.get(notification2.severity)) {
                    return -1;
                }
                if (priorityMap.get(notification1.severity) > priorityMap.get(notification2.severity)) {
                    return 1;
                }
                return _this.compareUpdateTime(new Date(notification1.updateTimestamp), new Date(notification2.updateTimestamp));
            }
        }));
    };
    /**
     * @return {?}
     */
    FxUserNotificationComponent.prototype.postReadNotificationLogAndRefreshNotifications = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.userNotificationService.postUserReadLog().toPromise().then((/**
         * @return {?}
         */
        function () {
            _this.receiveNotifications();
        }));
    };
    /**
     * @return {?}
     */
    FxUserNotificationComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        if (this.intervalTimeSubscription) {
            this.intervalTimeSubscription.unsubscribe();
        }
        if (this.userNotificationSubscription) {
            this.userNotificationSubscription.unsubscribe();
        }
    };
    /**
     * @private
     * @param {?} ts1
     * @param {?} ts2
     * @return {?}
     */
    FxUserNotificationComponent.prototype.compareUpdateTime = /**
     * @private
     * @param {?} ts1
     * @param {?} ts2
     * @return {?}
     */
    function (ts1, ts2) {
        if (ts1.getTime() < ts2.getTime()) {
            return 1;
        }
        else if (ts1.getTime() > ts2.getTime()) {
            return -1;
        }
        return 0;
    };
    FxUserNotificationComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-fx-user-notification',
                    template: "<fa-icon [icon]=\"icon\" [size]=\"size\" [ngStyle]=\"getNotificationIconColor()\" (click)=\"toggleNotificationArea()\" class=\"notification-icon\"></fa-icon>\r\n<span [ngClass]=\"notificationCount == 0 ? 'hide-count': 'show-count'\">{{ notificationCount }}</span>\r\n\r\n<div [ngClass]=\"opened ? 'show' : 'hide'\"\r\n     class=\"menu-content\">\r\n  <div class=\"cross-icon\" (click)=\"toggleNotificationArea()\">\r\n    <i class=\"fa fa-times closeIcon\"></i>\r\n  </div>\r\n  <h6 class=\"card-title\">{{'i18n.user-notifications.title' | translate }}</h6>\r\n  <div *ngFor=\"let notification of userNotificationList let i=index\">\r\n    <app-fx-user-single-notification-area\r\n      [notification]=\"notification\"\r\n      [charsOfShortText]=\"charsOfShortText\"\r\n    ></app-fx-user-single-notification-area>\r\n  </div>\r\n</div>\r\n\r\n\r\n\r\n",
                    styles: [".notification-card{max-width:500px;margin-top:24px}.hide-count{display:none}.menu-content{position:absolute;top:0;height:100vh;background-color:#f5f5f5}.show{display:block;position:absolute;overflow:auto;padding:8px;width:380px;margin-top:56px;height:calc(100vh - 96px)}.hide{display:none}.cross-icon{float:right;color:#fff!important}.icons{text-align:right}.closeIcon{color:grey;margin-right:5px;cursor:pointer}.show-count{margin-top:-1px;position:absolute;margin-left:-11px;color:#fff;background-color:red;min-width:21px;border-radius:6px;font-size:small;text-align:center}.card-title{background-color:#004666;color:#fff;text-align:center;height:23px}.notification-icon{cursor:pointer}"]
                }] }
    ];
    /** @nocollapse */
    FxUserNotificationComponent.ctorParameters = function () { return [
        { type: FxUserNotificationService }
    ]; };
    FxUserNotificationComponent.propDecorators = {
        icon: [{ type: Input }],
        color: [{ type: Input }],
        size: [{ type: Input }],
        refreshInterval: [{ type: Input }],
        charsOfShortText: [{ type: Input }]
    };
    return FxUserNotificationComponent;
}());
export { FxUserNotificationComponent };
if (false) {
    /** @type {?} */
    FxUserNotificationComponent.prototype.opened;
    /** @type {?} */
    FxUserNotificationComponent.prototype.icon;
    /** @type {?} */
    FxUserNotificationComponent.prototype.color;
    /** @type {?} */
    FxUserNotificationComponent.prototype.size;
    /** @type {?} */
    FxUserNotificationComponent.prototype.refreshInterval;
    /** @type {?} */
    FxUserNotificationComponent.prototype.charsOfShortText;
    /** @type {?} */
    FxUserNotificationComponent.prototype.userNotificationList;
    /** @type {?} */
    FxUserNotificationComponent.prototype.userNotificationUnreadList;
    /** @type {?} */
    FxUserNotificationComponent.prototype.userNotificationReadList;
    /** @type {?} */
    FxUserNotificationComponent.prototype.notificationCount;
    /** @type {?} */
    FxUserNotificationComponent.prototype.intervalTimeSubscription;
    /**
     * @type {?}
     * @private
     */
    FxUserNotificationComponent.prototype.userNotificationSubscription;
    /**
     * @type {?}
     * @private
     */
    FxUserNotificationComponent.prototype.userNotificationService;
}
