// @fx-lib/components/esm5/lib/fx-dropdown/fx-dropdown.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
var FxDropdownComponent = /** @class */ (function () {
    function FxDropdownComponent() {
        this.cssClass = '';
        this.filterable = false;
        this.currentItemChange = new EventEmitter();
    }
    /**
     * @return {?}
     */
    FxDropdownComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (!this.placeholder && this.options[0]) {
            this.currentItem = this.options[0];
        }
    };
    /**
     * @param {?} item
     * @return {?}
     */
    FxDropdownComponent.prototype.clickItem = /**
     * @param {?} item
     * @return {?}
     */
    function (item) {
        if (item && !item.disabled) {
            this.currentItem = item;
            if (item.onSelectAction) {
                item.onSelectAction();
            }
            this.currentItemChange.emit(item);
        }
    };
    /**
     * @param {?} item
     * @return {?}
     */
    FxDropdownComponent.prototype.getComputedLabel = /**
     * @param {?} item
     * @return {?}
     */
    function (item) {
        return item.label ? item.label : item.key;
    };
    /**
     * @param {?} item
     * @return {?}
     */
    FxDropdownComponent.prototype.isCurrentItem = /**
     * @param {?} item
     * @return {?}
     */
    function (item) {
        return item === this.currentItem;
    };
    /**
     * @param {?} item
     * @return {?}
     */
    FxDropdownComponent.prototype.isItemHighlighted = /**
     * @param {?} item
     * @return {?}
     */
    function (item) {
        return this.isCurrentItem(item) && this.highlightActive;
    };
    /**
     * @return {?}
     */
    FxDropdownComponent.prototype.getSelectedItem = /**
     * @return {?}
     */
    function () {
        return this.currentItem;
    };
    /**
     * @param {?} item
     * @return {?}
     */
    FxDropdownComponent.prototype.setSelectedItem = /**
     * @param {?} item
     * @return {?}
     */
    function (item) {
        this.currentItem = item;
        this.currentItemChange.emit(item);
    };
    /**
     * @param {?} searchString
     * @return {?}
     */
    FxDropdownComponent.prototype.setFilterText = /**
     * @param {?} searchString
     * @return {?}
     */
    function (searchString) {
        this.filterInput = searchString;
    };
    /**
     * @param {?} entries
     * @return {?}
     */
    FxDropdownComponent.prototype.getFilteredList = /**
     * @param {?} entries
     * @return {?}
     */
    function (entries) {
        var _this = this;
        if (this.filterable && this.filterInput && this.filterInput.length > 0) {
            return entries.filter((/**
             * @param {?} e
             * @return {?}
             */
            function (e) { return e.key.concat(e.label).toLowerCase().includes(_this.filterInput.toLowerCase()); }));
        }
        else {
            return entries;
        }
    };
    FxDropdownComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-fx-dropdown',
                    template: "<div ngbDropdown [ngClass]=\"['dropdown', cssClass]\">\r\n\r\n  <button ngbDropdownToggle class=\"btn btn-secondary cursor-type no-box-shadow\">\r\n    <span *ngIf=\"currentItem\">  {{ getComputedLabel(currentItem) }} </span>\r\n    <span *ngIf=\"!currentItem\">  {{ placeholder | translate}} </span>\r\n  </button>\r\n\r\n  <div ngbDropdownMenu class=\"dropdown-menu dropdown-primary\">\r\n    <app-fx-search-input *ngIf=\"filterable\" (searchInputChangedEvent)=\"setFilterText($event)\" placeholder=\"\"\r\n                         debounceTimeInMilliseconds=\"0\" searchFieldMinLength=\"1\" class=\"full-width\">\r\n    </app-fx-search-input>\r\n\r\n    <div class=\"dropdown-item-wrapper\">\r\n      <button ngbDropdownItem class=\"dropdown-item \" [class.active]=\"isItemHighlighted(item)\"\r\n              *ngFor=\"let item of getFilteredList(options)\"\r\n              (click)=\"clickItem(item)\"\r\n              [disabled]=\"item.disabled\">\r\n        {{getComputedLabel(item)}}\r\n      </button>\r\n    </div>\r\n  </div>\r\n</div>\r\n",
                    styles: [":host ::ng-deep app-fx-search-input .search-input{max-width:100%;position:-webkit-sticky;position:sticky;top:0;background:#fff;padding:5px}.dropdown-menu{padding:0}.dropdown-menu .dropdown-item-wrapper{max-height:50vh;overflow:auto}"]
                }] }
    ];
    FxDropdownComponent.propDecorators = {
        options: [{ type: Input }],
        placeholder: [{ type: Input }],
        highlightActive: [{ type: Input }],
        currentItem: [{ type: Input }],
        cssClass: [{ type: Input }],
        filterable: [{ type: Input }],
        currentItemChange: [{ type: Output }]
    };
    return FxDropdownComponent;
}());
export { FxDropdownComponent };
if (false) {
    /** @type {?} */
    FxDropdownComponent.prototype.options;
    /** @type {?} */
    FxDropdownComponent.prototype.placeholder;
    /** @type {?} */
    FxDropdownComponent.prototype.highlightActive;
    /** @type {?} */
    FxDropdownComponent.prototype.currentItem;
    /** @type {?} */
    FxDropdownComponent.prototype.cssClass;
    /** @type {?} */
    FxDropdownComponent.prototype.filterable;
    /** @type {?} */
    FxDropdownComponent.prototype.currentItemChange;
    /**
     * @type {?}
     * @private
     */
    FxDropdownComponent.prototype.filterInput;
}
