// @fx-lib/components/esm5/lib/shared/test/mocks/translate-service-mock.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { get } from 'lodash';
import { of, Subject } from 'rxjs';
/**
 * @record
 */
export function ITranslateServiceMock() { }
if (false) {
    /** @type {?} */
    ITranslateServiceMock.prototype.onTranslationChange;
    /** @type {?} */
    ITranslateServiceMock.prototype.onLangChange;
    /** @type {?} */
    ITranslateServiceMock.prototype.onDefaultLangChange;
    /**
     * @param {?} key
     * @param {?=} param
     * @return {?}
     */
    ITranslateServiceMock.prototype.get = function (key, param) { };
    /**
     * @param {?} key
     * @return {?}
     */
    ITranslateServiceMock.prototype.instant = function (key) { };
    /**
     * @return {?}
     */
    ITranslateServiceMock.prototype.getBrowserLang = function () { };
    /**
     * @return {?}
     */
    ITranslateServiceMock.prototype.getBrowserCultureLang = function () { };
    /**
     * @param {?} lang
     * @return {?}
     */
    ITranslateServiceMock.prototype.getTranslation = function (lang) { };
}
/**
 * @record
 */
export function ITranslateServiceMockConfig() { }
if (false) {
    /** @type {?|undefined} */
    ITranslateServiceMockConfig.prototype.browserCulterLang;
    /** @type {?|undefined} */
    ITranslateServiceMockConfig.prototype.currentLang;
    /** @type {?|undefined} */
    ITranslateServiceMockConfig.prototype.translation;
    /** @type {?|undefined} */
    ITranslateServiceMockConfig.prototype.translations;
}
var TranslateServiceMockImpl = /** @class */ (function () {
    function TranslateServiceMockImpl(_config) {
        this._config = _config;
        this.onLangChange = new Subject();
    }
    Object.defineProperty(TranslateServiceMockImpl.prototype, "currentLang", {
        get: /**
         * @return {?}
         */
        function () {
            return get(this._config, 'currentLang', this.getBrowserCultureLang().substr(0, 2));
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} key
     * @param {?=} param
     * @return {?}
     */
    TranslateServiceMockImpl.prototype.get = /**
     * @param {?} key
     * @param {?=} param
     * @return {?}
     */
    function (key, param) {
        var _this = this;
        if (Array.isArray(key)) {
            /** @type {?} */
            var keyMap_1 = {};
            key.forEach((/**
             * @param {?} k
             * @return {?}
             */
            function (k) {
                keyMap_1[k] = _this.instant(k);
            }));
            return of(keyMap_1);
        }
        return of(this.instant(key));
    };
    /**
     * @param {?} language
     * @return {?}
     */
    TranslateServiceMockImpl.prototype.switchLanguage = /**
     * @param {?} language
     * @return {?}
     */
    function (language) {
        this.onLangChange.next(language);
    };
    /**
     * @return {?}
     */
    TranslateServiceMockImpl.prototype.getBrowserCultureLang = /**
     * @return {?}
     */
    function () {
        return get(this._config, 'browserCultureLang', 'de-DE');
    };
    /**
     * @return {?}
     */
    TranslateServiceMockImpl.prototype.getBrowserLang = /**
     * @return {?}
     */
    function () {
        var _a = tslib_1.__read(this.getBrowserCultureLang().split('-'), 1), browserLang = _a[0];
        return browserLang;
    };
    /**
     * @param {?} lang
     * @return {?}
     */
    TranslateServiceMockImpl.prototype.getTranslation = /**
     * @param {?} lang
     * @return {?}
     */
    function (lang) {
        return of(get(this._config, "translation." + lang, lang));
    };
    /**
     * @param {?} key
     * @return {?}
     */
    TranslateServiceMockImpl.prototype.instant = /**
     * @param {?} key
     * @return {?}
     */
    function (key) {
        return get(this._config, "translations." + key, key);
    };
    return TranslateServiceMockImpl;
}());
export { TranslateServiceMockImpl };
if (false) {
    /** @type {?} */
    TranslateServiceMockImpl.prototype.onDefaultLangChange;
    /** @type {?} */
    TranslateServiceMockImpl.prototype.onLangChange;
    /** @type {?} */
    TranslateServiceMockImpl.prototype.onTranslationChange;
    /**
     * @type {?}
     * @private
     */
    TranslateServiceMockImpl.prototype._config;
}
/** @type {?} */
export var translateServiceMock = new TranslateServiceMockImpl({});
/**
 * @param {?} config
 * @return {?}
 */
export function createTranslateServiceMock(config) {
    return new TranslateServiceMockImpl(config);
}
