// @fx-lib/components/esm2015/lib/fx-user-notification/components/fx-user-notification/fx-user-notification.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { faBell } from '@fortawesome/free-solid-svg-icons';
import { FxUserNotificationService } from '../../services/fx-user-notification.service';
import { priorityMap } from '../../domain/fx-user-notification';
import { timer } from 'rxjs';
export class FxUserNotificationComponent {
    /**
     * @param {?} userNotificationService
     */
    constructor(userNotificationService) {
        this.userNotificationService = userNotificationService;
        this.opened = false;
        this.icon = faBell;
        this.color = 'white';
        this.size = '2x';
        this.refreshInterval = 60000;
        this.charsOfShortText = 50;
        this.userNotificationList = [];
        this.userNotificationUnreadList = [];
        this.userNotificationReadList = [];
    }
    /**
     * @return {?}
     */
    getNotificationIconColor() {
        return { color: this.color };
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.intervalTimeSubscription = timer(0, this.refreshInterval).subscribe((/**
         * @return {?}
         */
        () => {
            this.receiveNotifications();
        }));
    }
    /**
     * @return {?}
     */
    receiveNotifications() {
        this.userNotificationSubscription = this.userNotificationService.getNotifications().subscribe((/**
         * @param {?} res
         * @return {?}
         */
        res => {
            this.sortNotificationsAndRemember(res);
        }));
    }
    /**
     * @return {?}
     */
    toggleNotificationArea() {
        this.opened = !this.opened;
        if (!this.opened) {
            this.postReadNotificationLogAndRefreshNotifications();
        }
    }
    /**
     * @param {?} notificationData
     * @return {?}
     */
    sortNotificationsAndRemember(notificationData) {
        this.userNotificationReadList = notificationData.filter((/**
         * @param {?} notification
         * @return {?}
         */
        (notification) => {
            return notification.read;
        }));
        this.userNotificationUnreadList = notificationData.filter((/**
         * @param {?} notification
         * @return {?}
         */
        (notification) => {
            return !notification.read;
        }));
        this.userNotificationList = this.sortNotification(this.userNotificationUnreadList).concat(this.sortNotification(this.userNotificationReadList));
        this.notificationCount = this.userNotificationUnreadList.length;
    }
    /**
     * @param {?} notificationData
     * @return {?}
     */
    sortNotification(notificationData) {
        return notificationData.sort((/**
         * @param {?} notification1
         * @param {?} notification2
         * @return {?}
         */
        (notification1, notification2) => {
            {
                if (priorityMap.get(notification1.severity) < priorityMap.get(notification2.severity)) {
                    return -1;
                }
                if (priorityMap.get(notification1.severity) > priorityMap.get(notification2.severity)) {
                    return 1;
                }
                return this.compareUpdateTime(new Date(notification1.updateTimestamp), new Date(notification2.updateTimestamp));
            }
        }));
    }
    /**
     * @return {?}
     */
    postReadNotificationLogAndRefreshNotifications() {
        this.userNotificationService.postUserReadLog().toPromise().then((/**
         * @return {?}
         */
        () => {
            this.receiveNotifications();
        }));
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        if (this.intervalTimeSubscription) {
            this.intervalTimeSubscription.unsubscribe();
        }
        if (this.userNotificationSubscription) {
            this.userNotificationSubscription.unsubscribe();
        }
    }
    /**
     * @private
     * @param {?} ts1
     * @param {?} ts2
     * @return {?}
     */
    compareUpdateTime(ts1, ts2) {
        if (ts1.getTime() < ts2.getTime()) {
            return 1;
        }
        else if (ts1.getTime() > ts2.getTime()) {
            return -1;
        }
        return 0;
    }
}
FxUserNotificationComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-fx-user-notification',
                template: "<fa-icon [icon]=\"icon\" [size]=\"size\" [ngStyle]=\"getNotificationIconColor()\" (click)=\"toggleNotificationArea()\" class=\"notification-icon\"></fa-icon>\r\n<span [ngClass]=\"notificationCount == 0 ? 'hide-count': 'show-count'\">{{ notificationCount }}</span>\r\n\r\n<div [ngClass]=\"opened ? 'show' : 'hide'\"\r\n     class=\"menu-content\">\r\n  <div class=\"cross-icon\" (click)=\"toggleNotificationArea()\">\r\n    <i class=\"fa fa-times closeIcon\"></i>\r\n  </div>\r\n  <h6 class=\"card-title\">{{'i18n.user-notifications.title' | translate }}</h6>\r\n  <div *ngFor=\"let notification of userNotificationList let i=index\">\r\n    <app-fx-user-single-notification-area\r\n      [notification]=\"notification\"\r\n      [charsOfShortText]=\"charsOfShortText\"\r\n    ></app-fx-user-single-notification-area>\r\n  </div>\r\n</div>\r\n\r\n\r\n\r\n",
                styles: [".notification-card{max-width:500px;margin-top:24px}.hide-count{display:none}.menu-content{position:absolute;top:0;height:100vh;background-color:#f5f5f5}.show{display:block;position:absolute;overflow:auto;padding:8px;width:380px;margin-top:56px;height:calc(100vh - 96px)}.hide{display:none}.cross-icon{float:right;color:#fff!important}.icons{text-align:right}.closeIcon{color:grey;margin-right:5px;cursor:pointer}.show-count{margin-top:-1px;position:absolute;margin-left:-11px;color:#fff;background-color:red;min-width:21px;border-radius:6px;font-size:small;text-align:center}.card-title{background-color:#004666;color:#fff;text-align:center;height:23px}.notification-icon{cursor:pointer}"]
            }] }
];
/** @nocollapse */
FxUserNotificationComponent.ctorParameters = () => [
    { type: FxUserNotificationService }
];
FxUserNotificationComponent.propDecorators = {
    icon: [{ type: Input }],
    color: [{ type: Input }],
    size: [{ type: Input }],
    refreshInterval: [{ type: Input }],
    charsOfShortText: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    FxUserNotificationComponent.prototype.opened;
    /** @type {?} */
    FxUserNotificationComponent.prototype.icon;
    /** @type {?} */
    FxUserNotificationComponent.prototype.color;
    /** @type {?} */
    FxUserNotificationComponent.prototype.size;
    /** @type {?} */
    FxUserNotificationComponent.prototype.refreshInterval;
    /** @type {?} */
    FxUserNotificationComponent.prototype.charsOfShortText;
    /** @type {?} */
    FxUserNotificationComponent.prototype.userNotificationList;
    /** @type {?} */
    FxUserNotificationComponent.prototype.userNotificationUnreadList;
    /** @type {?} */
    FxUserNotificationComponent.prototype.userNotificationReadList;
    /** @type {?} */
    FxUserNotificationComponent.prototype.notificationCount;
    /** @type {?} */
    FxUserNotificationComponent.prototype.intervalTimeSubscription;
    /**
     * @type {?}
     * @private
     */
    FxUserNotificationComponent.prototype.userNotificationSubscription;
    /**
     * @type {?}
     * @private
     */
    FxUserNotificationComponent.prototype.userNotificationService;
}
