// @fx-lib/components/esm2015/lib/shared/test/mocks/activated-route-mock.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { BehaviorSubject } from 'rxjs';
export class ActivatedRouteMock {
    constructor() {
        this.fragment = new BehaviorSubject(this.testFragment);
        this.params = new BehaviorSubject(this.testParams);
        this.queryParams = new BehaviorSubject(this.testQueryParams);
    }
    /**
     * @return {?}
     */
    get testParams() {
        return this._testParams;
    }
    /**
     * @param {?} params
     * @return {?}
     */
    set testParams(params) {
        this._testParams = params;
        this.params.next(params);
    }
    /**
     * @return {?}
     */
    get testFragment() {
        return this._fragment;
    }
    /**
     * @param {?} fragment
     * @return {?}
     */
    set testFragment(fragment) {
        this._fragment = fragment;
        this.fragment.next(fragment);
    }
    /**
     * @return {?}
     */
    get testQueryParams() {
        return this._queryParams;
    }
    /**
     * @param {?} queryParams
     * @return {?}
     */
    set testQueryParams(queryParams) {
        this._queryParams = queryParams;
        this.queryParams.next(queryParams);
    }
    /**
     * @return {?}
     */
    get snapshot() {
        return {
            params: this.testParams,
            queryParams: this.testQueryParams,
            fragment: this.testFragment,
        };
    }
    /**
     * @return {?}
     */
    get parent() {
        return {
            fragment: this.fragment,
            params: this.params,
            queryParams: this.queryParams,
        };
    }
}
if (false) {
    /** @type {?} */
    ActivatedRouteMock.prototype.fragment;
    /** @type {?} */
    ActivatedRouteMock.prototype.params;
    /** @type {?} */
    ActivatedRouteMock.prototype.queryParams;
    /**
     * @type {?}
     * @private
     */
    ActivatedRouteMock.prototype._testParams;
    /**
     * @type {?}
     * @private
     */
    ActivatedRouteMock.prototype._fragment;
    /**
     * @type {?}
     * @private
     */
    ActivatedRouteMock.prototype._queryParams;
}
