// @fx-lib/components/esm2015/lib/shared/test/mocks/translate-service-mock.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { get } from 'lodash';
import { of, Subject } from 'rxjs';
/**
 * @record
 */
export function ITranslateServiceMock() { }
if (false) {
    /** @type {?} */
    ITranslateServiceMock.prototype.onTranslationChange;
    /** @type {?} */
    ITranslateServiceMock.prototype.onLangChange;
    /** @type {?} */
    ITranslateServiceMock.prototype.onDefaultLangChange;
    /**
     * @param {?} key
     * @param {?=} param
     * @return {?}
     */
    ITranslateServiceMock.prototype.get = function (key, param) { };
    /**
     * @param {?} key
     * @return {?}
     */
    ITranslateServiceMock.prototype.instant = function (key) { };
    /**
     * @return {?}
     */
    ITranslateServiceMock.prototype.getBrowserLang = function () { };
    /**
     * @return {?}
     */
    ITranslateServiceMock.prototype.getBrowserCultureLang = function () { };
    /**
     * @param {?} lang
     * @return {?}
     */
    ITranslateServiceMock.prototype.getTranslation = function (lang) { };
}
/**
 * @record
 */
export function ITranslateServiceMockConfig() { }
if (false) {
    /** @type {?|undefined} */
    ITranslateServiceMockConfig.prototype.browserCulterLang;
    /** @type {?|undefined} */
    ITranslateServiceMockConfig.prototype.currentLang;
    /** @type {?|undefined} */
    ITranslateServiceMockConfig.prototype.translation;
    /** @type {?|undefined} */
    ITranslateServiceMockConfig.prototype.translations;
}
export class TranslateServiceMockImpl {
    /**
     * @param {?} _config
     */
    constructor(_config) {
        this._config = _config;
        this.onLangChange = new Subject();
    }
    /**
     * @return {?}
     */
    get currentLang() {
        return get(this._config, 'currentLang', this.getBrowserCultureLang().substr(0, 2));
    }
    /**
     * @param {?} key
     * @param {?=} param
     * @return {?}
     */
    get(key, param) {
        if (Array.isArray(key)) {
            /** @type {?} */
            const keyMap = {};
            key.forEach((/**
             * @param {?} k
             * @return {?}
             */
            k => {
                keyMap[k] = this.instant(k);
            }));
            return of(keyMap);
        }
        return of(this.instant(key));
    }
    /**
     * @param {?} language
     * @return {?}
     */
    switchLanguage(language) {
        this.onLangChange.next(language);
    }
    /**
     * @return {?}
     */
    getBrowserCultureLang() {
        return get(this._config, 'browserCultureLang', 'de-DE');
    }
    /**
     * @return {?}
     */
    getBrowserLang() {
        const [browserLang] = this.getBrowserCultureLang().split('-');
        return browserLang;
    }
    /**
     * @param {?} lang
     * @return {?}
     */
    getTranslation(lang) {
        return of(get(this._config, `translation.${lang}`, lang));
    }
    /**
     * @param {?} key
     * @return {?}
     */
    instant(key) {
        return get(this._config, `translations.${key}`, key);
    }
}
if (false) {
    /** @type {?} */
    TranslateServiceMockImpl.prototype.onDefaultLangChange;
    /** @type {?} */
    TranslateServiceMockImpl.prototype.onLangChange;
    /** @type {?} */
    TranslateServiceMockImpl.prototype.onTranslationChange;
    /**
     * @type {?}
     * @private
     */
    TranslateServiceMockImpl.prototype._config;
}
/** @type {?} */
export const translateServiceMock = new TranslateServiceMockImpl({});
/**
 * @param {?} config
 * @return {?}
 */
export function createTranslateServiceMock(config) {
    return new TranslateServiceMockImpl(config);
}
