// @fx-lib/components/esm5/lib/fx-boundary-selector/services/boundary-data.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable, Inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { AuthenticationService } from '../../fx-security/services/authentication.service';
import { SECURITY_CONFIG } from '../../fx-security/domain/security-config';
import { AuthorizationService } from '../../fx-security/services/authorization.service';
import { map } from 'rxjs/operators';
var BoundaryDataService = /** @class */ (function () {
    function BoundaryDataService(http, authenticationService, authorizationService, securityConfig) {
        this.http = http;
        this.authenticationService = authenticationService;
        this.authorizationService = authorizationService;
        this.securityConfig = securityConfig;
    }
    /**
     * @return {?}
     */
    BoundaryDataService.prototype.getBoundariesForUserScope = /**
     * @return {?}
     */
    function () {
        var _this = this;
        return this.http.get(this.getApiUrl()).pipe(map((/**
         * @param {?} boundarySelectorResources
         * @return {?}
         */
        function (boundarySelectorResources) {
            return _this.filterBoundariesForUserScope(boundarySelectorResources);
        })));
    };
    /**
     * @param {?} boundarySelectorResources
     * @return {?}
     */
    BoundaryDataService.prototype.filterBoundariesForUserScope = /**
     * @param {?} boundarySelectorResources
     * @return {?}
     */
    function (boundarySelectorResources) {
        var _this = this;
        /** @type {?} */
        var filteredBoundarySelectorResources = [];
        if (this.authorizationService.hasDevopsRole()) {
            return boundarySelectorResources;
        }
        this.authenticationService.getPayload().roles.forEach((/**
         * @param {?} role
         * @return {?}
         */
        function (role) {
            if (role.boundaries[_this.securityConfig.boundary_name]) {
                role.boundaries[_this.securityConfig.boundary_name].forEach((/**
                 * @param {?} scope
                 * @return {?}
                 */
                function (scope) {
                    /** @type {?} */
                    var boundaryFound = boundarySelectorResources.find((/**
                     * @param {?} boundary
                     * @return {?}
                     */
                    function (boundary) { return boundary.boundaryValueKey === scope; }));
                    if (boundaryFound) {
                        filteredBoundarySelectorResources.push(boundaryFound);
                    }
                }));
            }
        }));
        return filteredBoundarySelectorResources;
    };
    /**
     * @private
     * @return {?}
     */
    BoundaryDataService.prototype.getApiUrl = /**
     * @private
     * @return {?}
     */
    function () {
        return this.securityConfig.authorization_service_url + "/api/v1/domains/" + this.securityConfig.app_domain + "/apps/" + this.securityConfig.app_name + "/boundaryvaluesminified/";
    };
    BoundaryDataService.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    BoundaryDataService.ctorParameters = function () { return [
        { type: HttpClient },
        { type: AuthenticationService },
        { type: AuthorizationService },
        { type: undefined, decorators: [{ type: Inject, args: [SECURITY_CONFIG,] }] }
    ]; };
    return BoundaryDataService;
}());
export { BoundaryDataService };
if (false) {
    /** @type {?} */
    BoundaryDataService.prototype.securityConfig;
    /**
     * @type {?}
     * @private
     */
    BoundaryDataService.prototype.http;
    /**
     * @type {?}
     * @private
     */
    BoundaryDataService.prototype.authenticationService;
    /**
     * @type {?}
     * @private
     */
    BoundaryDataService.prototype.authorizationService;
}
