// @fx-lib/components/esm2015/lib/fx-security/components/idp-login/idp-login.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component } from '@angular/core';
import { Router } from '@angular/router';
import { NotificationsService } from 'angular2-notifications';
import { NEVER, Subscription } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { AuthenticationService } from '../../services/authentication.service';
export class IdpLoginComponent {
    /**
     * @param {?} authenticationService
     * @param {?} router
     * @param {?} notificationService
     */
    constructor(authenticationService, router, notificationService) {
        this.authenticationService = authenticationService;
        this.router = router;
        this.notificationService = notificationService;
        this.subscription = new Subscription();
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.subscription.add(this.authenticationService.loginFromUrlToken(window.location.href)
            .pipe(catchError((/**
         * @param {?} error
         * @return {?}
         */
        error => {
            this.notificationService.error(error);
            return NEVER;
        })))
            .subscribe((/**
         * @param {?} identityToken
         * @return {?}
         */
        (identityToken) => {
            this.router.navigateByUrl(identityToken.state);
        })));
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        this.subscription.unsubscribe();
    }
}
IdpLoginComponent.decorators = [
    { type: Component, args: [{
                template: ''
            }] }
];
/** @nocollapse */
IdpLoginComponent.ctorParameters = () => [
    { type: AuthenticationService },
    { type: Router },
    { type: NotificationsService }
];
if (false) {
    /**
     * @type {?}
     * @private
     */
    IdpLoginComponent.prototype.subscription;
    /**
     * @type {?}
     * @private
     */
    IdpLoginComponent.prototype.authenticationService;
    /**
     * @type {?}
     * @private
     */
    IdpLoginComponent.prototype.router;
    /**
     * @type {?}
     * @private
     */
    IdpLoginComponent.prototype.notificationService;
}
