// @fx-lib/components/esm5/lib/fx-header/services/fx-header-config.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import { FxHeaderConfigValidatorService } from './fx-header-config-validator.service';
var FxHeaderConfigService = /** @class */ (function () {
    function FxHeaderConfigService(fxheaderConfigValidatorService) {
        this.fxheaderConfigValidatorService = fxheaderConfigValidatorService;
        this.config = new BehaviorSubject({
            appTitleShort: '',
            appTitle: '',
            routes: [],
            staticComponents: [],
            components: []
        });
    }
    /**
     * @param {?} config
     * @return {?}
     */
    FxHeaderConfigService.prototype.setConfig = /**
     * @param {?} config
     * @return {?}
     */
    function (config) {
        if (!this.fxheaderConfigValidatorService.validate(config)) {
            return;
        }
        this.config.next(config);
    };
    FxHeaderConfigService.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    FxHeaderConfigService.ctorParameters = function () { return [
        { type: FxHeaderConfigValidatorService }
    ]; };
    return FxHeaderConfigService;
}());
export { FxHeaderConfigService };
if (false) {
    /** @type {?} */
    FxHeaderConfigService.prototype.config;
    /**
     * @type {?}
     * @private
     */
    FxHeaderConfigService.prototype.fxheaderConfigValidatorService;
}
