// @fx-lib/components/esm2015/lib/fx-user-notification/fx-user-notification.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FxUserNotificationComponent } from './components/fx-user-notification/fx-user-notification.component';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { BrowserModule } from '@angular/platform-browser';
import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
import { FormsModule } from '@angular/forms';
import { NOTIFICATION_CONFIG } from './domain/fx-notification-config';
import { FxTranslationModule } from '../fx-translation/fx-translation.module';
import { FxUserNotificationService } from './services/fx-user-notification.service';
import { FxUserNotificationLocalDevelopmentService } from './services/fx-user-notification-local-development.service';
import { FxUserSingleNotificationAreaComponent } from './components/fx-user-single-notification-area/fx-user-single-notification-area.component';
export class FxUserNotificationModule {
    /**
     * @param {?} notificationConfig
     * @return {?}
     */
    static forRoot(notificationConfig) {
        return {
            ngModule: FxUserNotificationModule,
            providers: [{
                    provide: NOTIFICATION_CONFIG,
                    useValue: notificationConfig
                },
                FxUserNotificationService]
        };
    }
    /**
     * @param {?} notificationConfig
     * @return {?}
     */
    static forLocalDevelopment(notificationConfig) {
        return {
            ngModule: FxUserNotificationModule,
            providers: [
                {
                    provide: NOTIFICATION_CONFIG,
                    useValue: notificationConfig
                },
                {
                    provide: FxUserNotificationService,
                    useClass: FxUserNotificationLocalDevelopmentService
                }
            ]
        };
    }
}
FxUserNotificationModule.decorators = [
    { type: NgModule, args: [{
                declarations: [FxUserNotificationComponent, FxUserSingleNotificationAreaComponent],
                exports: [
                    FxUserNotificationComponent,
                    FxUserSingleNotificationAreaComponent
                ],
                imports: [
                    BrowserModule,
                    CommonModule,
                    NgbModule,
                    FontAwesomeModule,
                    FormsModule,
                    FxTranslationModule
                ],
                bootstrap: [FxUserNotificationComponent]
            },] }
];
/**
 * @record
 */
export function FxNotificationConfig() { }
if (false) {
    /** @type {?} */
    FxNotificationConfig.prototype.notification_service_url;
}
