// @fx-lib/components/esm2015/lib/fx-dropdown/fx-dropdown.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
export class FxDropdownComponent {
    constructor() {
        this.cssClass = '';
        this.filterable = false;
        this.currentItemChange = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        if (!this.placeholder && this.options[0]) {
            this.currentItem = this.options[0];
        }
    }
    /**
     * @param {?} item
     * @return {?}
     */
    clickItem(item) {
        if (item && !item.disabled) {
            this.currentItem = item;
            if (item.onSelectAction) {
                item.onSelectAction();
            }
            this.currentItemChange.emit(item);
        }
    }
    /**
     * @param {?} item
     * @return {?}
     */
    getComputedLabel(item) {
        return item.label ? item.label : item.key;
    }
    /**
     * @param {?} item
     * @return {?}
     */
    isCurrentItem(item) {
        return item === this.currentItem;
    }
    /**
     * @param {?} item
     * @return {?}
     */
    isItemHighlighted(item) {
        return this.isCurrentItem(item) && this.highlightActive;
    }
    /**
     * @return {?}
     */
    getSelectedItem() {
        return this.currentItem;
    }
    /**
     * @param {?} item
     * @return {?}
     */
    setSelectedItem(item) {
        this.currentItem = item;
        this.currentItemChange.emit(item);
    }
    /**
     * @param {?} searchString
     * @return {?}
     */
    setFilterText(searchString) {
        this.filterInput = searchString;
    }
    /**
     * @param {?} entries
     * @return {?}
     */
    getFilteredList(entries) {
        if (this.filterable && this.filterInput && this.filterInput.length > 0) {
            return entries.filter((/**
             * @param {?} e
             * @return {?}
             */
            e => e.key.concat(e.label).toLowerCase().includes(this.filterInput.toLowerCase())));
        }
        else {
            return entries;
        }
    }
}
FxDropdownComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-fx-dropdown',
                template: "<div ngbDropdown [ngClass]=\"['dropdown', cssClass]\">\r\n\r\n  <button ngbDropdownToggle class=\"btn btn-secondary cursor-type no-box-shadow\">\r\n    <span *ngIf=\"currentItem\">  {{ getComputedLabel(currentItem) }} </span>\r\n    <span *ngIf=\"!currentItem\">  {{ placeholder | translate}} </span>\r\n  </button>\r\n\r\n  <div ngbDropdownMenu class=\"dropdown-menu dropdown-primary\">\r\n    <app-fx-search-input *ngIf=\"filterable\" (searchInputChangedEvent)=\"setFilterText($event)\" placeholder=\"\"\r\n                         debounceTimeInMilliseconds=\"0\" searchFieldMinLength=\"1\" class=\"full-width\">\r\n    </app-fx-search-input>\r\n\r\n    <div class=\"dropdown-item-wrapper\">\r\n      <button ngbDropdownItem class=\"dropdown-item \" [class.active]=\"isItemHighlighted(item)\"\r\n              *ngFor=\"let item of getFilteredList(options)\"\r\n              (click)=\"clickItem(item)\"\r\n              [disabled]=\"item.disabled\">\r\n        {{getComputedLabel(item)}}\r\n      </button>\r\n    </div>\r\n  </div>\r\n</div>\r\n",
                styles: [":host ::ng-deep app-fx-search-input .search-input{max-width:100%;position:-webkit-sticky;position:sticky;top:0;background:#fff;padding:5px}.dropdown-menu{padding:0}.dropdown-menu .dropdown-item-wrapper{max-height:50vh;overflow:auto}"]
            }] }
];
FxDropdownComponent.propDecorators = {
    options: [{ type: Input }],
    placeholder: [{ type: Input }],
    highlightActive: [{ type: Input }],
    currentItem: [{ type: Input }],
    cssClass: [{ type: Input }],
    filterable: [{ type: Input }],
    currentItemChange: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    FxDropdownComponent.prototype.options;
    /** @type {?} */
    FxDropdownComponent.prototype.placeholder;
    /** @type {?} */
    FxDropdownComponent.prototype.highlightActive;
    /** @type {?} */
    FxDropdownComponent.prototype.currentItem;
    /** @type {?} */
    FxDropdownComponent.prototype.cssClass;
    /** @type {?} */
    FxDropdownComponent.prototype.filterable;
    /** @type {?} */
    FxDropdownComponent.prototype.currentItemChange;
    /**
     * @type {?}
     * @private
     */
    FxDropdownComponent.prototype.filterInput;
}
