// @fx-lib/components/esm2015/lib/fx-week-selector/fx-week-selector-i18n.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { NgbDatepickerI18n } from '@ng-bootstrap/ng-bootstrap';
import { TranslateService } from '@ngx-translate/core';
/** @type {?} */
const I18N_VALUES = {
    de: {
        weekdays: ['Mo', 'Di', 'Mi', 'Do', 'Fr', 'Sa', 'So'],
        months_long: ['Januar', 'Februar', 'März', 'April', 'Mai', 'Juni', 'Juli', 'August', 'September', 'Oktober', 'November', 'Dezember'],
        months_short: ['Jan', 'Feb', 'Mär', 'Apr', 'Mai', 'Jun', 'Jul', 'Aug', 'Sep', 'Okt', 'Nov', 'Dez'],
    },
    en: {
        weekdays: ['Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa', 'Su'],
        months_long: ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'],
        months_short: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dez'],
    }
};
export class I18n {
    /**
     * @param {?} translateService
     */
    constructor(translateService) {
        this.translateService = translateService;
        this.language = 'en';
        translateService.onLangChange.subscribe((/**
         * @param {?} langChangeEvent
         * @return {?}
         */
        (langChangeEvent) => {
            this.language = langChangeEvent.lang;
        }));
    }
}
I18n.decorators = [
    { type: Injectable }
];
/** @nocollapse */
I18n.ctorParameters = () => [
    { type: TranslateService }
];
if (false) {
    /** @type {?} */
    I18n.prototype.language;
    /**
     * @type {?}
     * @private
     */
    I18n.prototype.translateService;
}
export class FxWeekSelectorI18n extends NgbDatepickerI18n {
    /**
     * @param {?} i18n
     */
    constructor(i18n) {
        super();
        this.i18n = i18n;
    }
    /**
     * @param {?} date
     * @return {?}
     */
    getDayAriaLabel(date) {
        return `${date.year}-${date.month}-${date.day}`;
    }
    /**
     * @param {?} month
     * @return {?}
     */
    getMonthFullName(month) {
        return I18N_VALUES[this.i18n.language].months_long[month - 1];
    }
    /**
     * @param {?} month
     * @return {?}
     */
    getMonthShortName(month) {
        return this.getMonthFullName(month);
    }
    /**
     * @param {?} weekday
     * @return {?}
     */
    getWeekdayShortName(weekday) {
        return I18N_VALUES[this.i18n.language].weekdays[weekday - 1];
    }
}
FxWeekSelectorI18n.decorators = [
    { type: Injectable }
];
/** @nocollapse */
FxWeekSelectorI18n.ctorParameters = () => [
    { type: I18n }
];
if (false) {
    /**
     * @type {?}
     * @private
     */
    FxWeekSelectorI18n.prototype.i18n;
}
