// @fx-lib/components/esm5/lib/fx-week-selector/fx-week-selector-i18n.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@angular/core';
import { NgbDatepickerI18n } from '@ng-bootstrap/ng-bootstrap';
import { TranslateService } from '@ngx-translate/core';
/** @type {?} */
var I18N_VALUES = {
    de: {
        weekdays: ['Mo', 'Di', 'Mi', 'Do', 'Fr', 'Sa', 'So'],
        months_long: ['Januar', 'Februar', 'März', 'April', 'Mai', 'Juni', 'Juli', 'August', 'September', 'Oktober', 'November', 'Dezember'],
        months_short: ['Jan', 'Feb', 'Mär', 'Apr', 'Mai', 'Jun', 'Jul', 'Aug', 'Sep', 'Okt', 'Nov', 'Dez'],
    },
    en: {
        weekdays: ['Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa', 'Su'],
        months_long: ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'],
        months_short: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dez'],
    }
};
var I18n = /** @class */ (function () {
    function I18n(translateService) {
        var _this = this;
        this.translateService = translateService;
        this.language = 'en';
        translateService.onLangChange.subscribe((/**
         * @param {?} langChangeEvent
         * @return {?}
         */
        function (langChangeEvent) {
            _this.language = langChangeEvent.lang;
        }));
    }
    I18n.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    I18n.ctorParameters = function () { return [
        { type: TranslateService }
    ]; };
    return I18n;
}());
export { I18n };
if (false) {
    /** @type {?} */
    I18n.prototype.language;
    /**
     * @type {?}
     * @private
     */
    I18n.prototype.translateService;
}
var FxWeekSelectorI18n = /** @class */ (function (_super) {
    tslib_1.__extends(FxWeekSelectorI18n, _super);
    function FxWeekSelectorI18n(i18n) {
        var _this = _super.call(this) || this;
        _this.i18n = i18n;
        return _this;
    }
    /**
     * @param {?} date
     * @return {?}
     */
    FxWeekSelectorI18n.prototype.getDayAriaLabel = /**
     * @param {?} date
     * @return {?}
     */
    function (date) {
        return date.year + "-" + date.month + "-" + date.day;
    };
    /**
     * @param {?} month
     * @return {?}
     */
    FxWeekSelectorI18n.prototype.getMonthFullName = /**
     * @param {?} month
     * @return {?}
     */
    function (month) {
        return I18N_VALUES[this.i18n.language].months_long[month - 1];
    };
    /**
     * @param {?} month
     * @return {?}
     */
    FxWeekSelectorI18n.prototype.getMonthShortName = /**
     * @param {?} month
     * @return {?}
     */
    function (month) {
        return this.getMonthFullName(month);
    };
    /**
     * @param {?} weekday
     * @return {?}
     */
    FxWeekSelectorI18n.prototype.getWeekdayShortName = /**
     * @param {?} weekday
     * @return {?}
     */
    function (weekday) {
        return I18N_VALUES[this.i18n.language].weekdays[weekday - 1];
    };
    FxWeekSelectorI18n.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    FxWeekSelectorI18n.ctorParameters = function () { return [
        { type: I18n }
    ]; };
    return FxWeekSelectorI18n;
}(NgbDatepickerI18n));
export { FxWeekSelectorI18n };
if (false) {
    /**
     * @type {?}
     * @private
     */
    FxWeekSelectorI18n.prototype.i18n;
}
