// @fx-lib/components/esm2015/lib/fx-user-notification/components/fx-user-single-notification-area/fx-user-single-notification-area.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class FxUserSingleNotificationAreaComponent {
    constructor() {
        this.charsOfShortText = 150;
        this.isMessageShorted = false;
        this.visibleMessage = '';
        this.show = false;
    }
    /**
     * @param {?} notification
     * @return {?}
     */
    set notification(notification) {
        this.completeNotification = notification;
        this.updateVisibleMessageText();
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.updateVisibleMessageText();
    }
    /**
     * @param {?} notification
     * @return {?}
     */
    getSeverityStyle(notification) {
        if (!notification) {
            return '';
        }
        if (!notification.read) {
            if (notification.severity === 'ERROR') {
                return 'alert alert-danger';
            }
            else if (notification.severity === 'WARNING') {
                return 'alert alert-warning';
            }
            else {
                return 'alert alert-info';
            }
        }
        else {
            return 'alert alert-secondary';
        }
    }
    /**
     * @return {?}
     */
    showToggle() {
        this.show = !this.show;
        this.updateVisibleMessageText();
    }
    /**
     * @param {?} notification
     * @return {?}
     */
    getSeverityIcon(notification) {
        if (!notification) {
            return '';
        }
        if (notification.severity === 'ERROR') {
            return 'fas fa-exclamation-circle';
        }
        else if (notification.severity === 'WARNING') {
            return 'fas fa-exclamation-triangle';
        }
        else {
            return 'fas fa-info-circle';
        }
    }
    /**
     * @private
     * @return {?}
     */
    updateVisibleMessageText() {
        if (!this.completeNotification) {
            return;
        }
        if (this.completeNotification.messageText.length > this.charsOfShortText && !this.show) {
            this.visibleMessage = this.completeNotification.messageText.slice(0, this.charsOfShortText);
            this.isMessageShorted = true;
        }
        else {
            this.visibleMessage = this.completeNotification.messageText;
        }
    }
}
FxUserSingleNotificationAreaComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-fx-user-single-notification-area',
                template: "<div [ngClass]=\"getSeverityStyle(completeNotification)\" class=\"container\">\r\n  <div>\r\n    <span class=\"msg-title\">{{completeNotification?.messageHeadline}}</span>\r\n    <span><i class=\"notificationIcon\" [ngClass]=\"getSeverityIcon(completeNotification)\"></i></span>\r\n  </div>\r\n  <div>\r\n    <span class=\"msg-datetime\">{{completeNotification?.updateTimestamp | date:'yyyy-MM-dd HH:mm'}}</span>\r\n  </div>\r\n  <div class=\"message-text\" [class.show-more]=\"show\">\r\n    {{ visibleMessage }}<span *ngIf=\"isMessageShorted && !show\">...</span>\r\n    <div class=\"showMoreLink\"\r\n         *ngIf=\"isMessageShorted\"\r\n    >\r\n      <a\r\n        (click)=\"showToggle()\"\r\n      >{{ (show ? 'i18n.user-notifications.showLess' : 'i18n.user-notifications.showMore') | translate}}</a>\r\n    </div>\r\n  </div>\r\n</div>\r\n",
                styles: [".notificationIcon{float:right;margin:.25em;font-size:xx-large}.showMoreLink{color:#003f5b;font-size:small;text-align:right;padding-top:5px}.showMoreLink a{cursor:pointer}.message-text{margin:.25em .25em .25em 0}.show-more{overflow:visible;word-wrap:break-spaces;white-space:normal}.msg-title{font-weight:700}.msg-datetime{font-size:smaller;margin-top:0}.container{padding:.25em;margin:0 0 .5em}"]
            }] }
];
/** @nocollapse */
FxUserSingleNotificationAreaComponent.ctorParameters = () => [];
FxUserSingleNotificationAreaComponent.propDecorators = {
    notification: [{ type: Input }],
    charsOfShortText: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    FxUserSingleNotificationAreaComponent.prototype.completeNotification;
    /** @type {?} */
    FxUserSingleNotificationAreaComponent.prototype.charsOfShortText;
    /** @type {?} */
    FxUserSingleNotificationAreaComponent.prototype.isMessageShorted;
    /** @type {?} */
    FxUserSingleNotificationAreaComponent.prototype.visibleMessage;
    /** @type {?} */
    FxUserSingleNotificationAreaComponent.prototype.show;
}
