// @fx-lib/components/esm2015/lib/fx-security/components/no-permissions/no-permissions.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Inject } from '@angular/core';
import { Router } from '@angular/router';
import { SECURITY_CONFIG } from '../../domain/security-config';
export class NoPermissionsComponent {
    /**
     * @param {?} router
     * @param {?} securityConfig
     */
    constructor(router, securityConfig) {
        this.router = router;
        this.securityConfig = securityConfig;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.ursLink = this.securityConfig.authorization_ui_url;
    }
}
NoPermissionsComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-no-permissions',
                template: "<div class=\"d-flex justify-content-center align-items-center\">\r\n    <div class=\"card text-center\">\r\n        <div class=\"card-header\">{{ 'i18n.fx-components.security.pages.no-permissions.header' | translate }}</div>\r\n        <div class=\"card-body\">\r\n            {{ 'i18n.fx-components.security.pages.no-permissions.text' | translate }}\r\n            <br>\r\n            <a [href]=\"ursLink\">{{ ursLink }}</a>\r\n        </div>\r\n    </div>\r\n</div>",
                styles: [".card{max-width:800px;margin-top:20%}"]
            }] }
];
/** @nocollapse */
NoPermissionsComponent.ctorParameters = () => [
    { type: Router },
    { type: undefined, decorators: [{ type: Inject, args: [SECURITY_CONFIG,] }] }
];
if (false) {
    /** @type {?} */
    NoPermissionsComponent.prototype.ursLink;
    /**
     * @type {?}
     * @protected
     */
    NoPermissionsComponent.prototype.securityConfig;
    /**
     * @type {?}
     * @private
     */
    NoPermissionsComponent.prototype.router;
}
