// @fx-lib/components/esm5/lib/fx-notification/fx-notification.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FxNotificationComponent } from './fx-notification.component';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { SimpleNotificationsModule } from 'angular2-notifications';
import { FxNotificationService } from './fx-notification.service';
var FxNotificationModule = /** @class */ (function () {
    function FxNotificationModule() {
    }
    /**
     * @return {?}
     */
    FxNotificationModule.forRoot = /**
     * @return {?}
     */
    function () {
        return {
            ngModule: FxNotificationModule,
            providers: [FxNotificationService]
        };
    };
    FxNotificationModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [FxNotificationComponent],
                    exports: [
                        FxNotificationComponent
                    ],
                    imports: [
                        CommonModule,
                        BrowserModule,
                        BrowserAnimationsModule,
                        SimpleNotificationsModule.forRoot()
                    ]
                },] }
    ];
    return FxNotificationModule;
}());
export { FxNotificationModule };
